%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english,openany,oneside]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
 \ifdefined\DeclareUnicodeCharacterAsOptional
  \DeclareUnicodeCharacter{"00A0}{\nobreakspace}
  \DeclareUnicodeCharacter{"2500}{\sphinxunichar{2500}}
  \DeclareUnicodeCharacter{"2502}{\sphinxunichar{2502}}
  \DeclareUnicodeCharacter{"2514}{\sphinxunichar{2514}}
  \DeclareUnicodeCharacter{"251C}{\sphinxunichar{251C}}
  \DeclareUnicodeCharacter{"2572}{\textbackslash}
 \else
  \DeclareUnicodeCharacter{00A0}{\nobreakspace}
  \DeclareUnicodeCharacter{2500}{\sphinxunichar{2500}}
  \DeclareUnicodeCharacter{2502}{\sphinxunichar{2502}}
  \DeclareUnicodeCharacter{2514}{\sphinxunichar{2514}}
  \DeclareUnicodeCharacter{251C}{\sphinxunichar{251C}}
  \DeclareUnicodeCharacter{2572}{\textbackslash}
 \fi
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\usepackage{geometry}

% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\figurename}{Fig.}}
\addto\captionsenglish{\renewcommand{\tablename}{Table}}
\addto\captionsenglish{\renewcommand{\literalblockname}{Listing}}

\addto\captionsenglish{\renewcommand{\literalblockcontinuedname}{continued from previous page}}
\addto\captionsenglish{\renewcommand{\literalblockcontinuesname}{continues on next page}}

\addto\extrasenglish{\def\pageautorefname{page}}

\setcounter{tocdepth}{1}



\title{El-MAVEN Documentation}
\date{Oct 18, 2018}
\release{v0.4.1}
\author{Elucidata Corporation}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex

\begin{document}

\maketitle
\sphinxtableofcontents
\phantomsection\label{\detokenize{index::doc}}


An intuitive, opensource LC-MS data processing engine from
\sphinxhref{http://www.elucidata.io/}{\sphinxincludegraphics{{Elucidata}.png}}
\begin{DUlineblock}{0em}
\item[] 
\end{DUlineblock}

\sphinxstylestrong{Travis Build}
\sphinxhref{https://travis-ci.org/ElucidataInc/ElMaven}{\sphinxincludegraphics{{ElMaven}.png}}
\begin{DUlineblock}{0em}
\item[] 
\end{DUlineblock}

\sphinxstylestrong{Digital Object Identifier}
\sphinxhref{https://doi.org/10.5281/zenodo.1332034}{\sphinxincludegraphics{{zenodo.1332034}.png}}

\chapter{Contents}
\label{\detokenize{index:contents}}

\section{El-MAVEN Features}
\label{\detokenize{ElMAVENFeatures:el-maven-features}}\label{\detokenize{ElMAVENFeatures::doc}}
\sphinxhref{http://genomics-pubs.princeton.edu/mzroll/index.php}{MAVEN} and \sphinxhref{https://elucidatainc.github.io/ElMaven/}{El-MAVEN} share the following features:
\begin{itemize}
\item {} 
Multi-file chromatographic aligner

\item {} 
Peak-feature detector

\item {} 
Isotope and adduct calculator

\item {} 
Formula predictor

\item {} 
Pathway visualizer

\item {} 
Isotopic flux animator

\end{itemize}

El-MAVEN is faster, more robust and has more user friendly features as compared to MAVEN.


\section{Download}
\label{\detokenize{Download:download}}\label{\detokenize{Download::doc}}
El-MAVEN installers are available for Windows (7, 8, 10) and MacOS (1.5 and above). Download \sphinxhref{https://elucidatainc.github.io/ElMaven/}{El-MAVEN’s} latest version or daily build for your preferred environment.


\section{Build}
\label{\detokenize{Build:build}}\label{\detokenize{Build::doc}}
Contributors can build El-MAVEN on Windows, Ubuntu or Mac systems by following these instructions.
Users are recommended to download the installers provided \sphinxhref{https://elucidatainc.github.io/ElMaven/Downloads/}{here}.


\subsection{Windows}
\label{\detokenize{Build:windows}}\begin{itemize}
\item {} 
Download \sphinxhref{http://www.msys2.org/}{MSYS2} installer and follow the installation instructions provided on their website.

\item {} 
Open MSYS2 and give the following commands to set up libraries and tool chains for El-MAVEN. Reopen MSYS2 when required:
\begin{quote}
\begin{itemize}
\item {} 
For 64 bit

\end{itemize}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} pacman \PYGZhy{}\PYGZhy{}force \PYGZhy{}Sy
\PYGZdl{} pacman \PYGZhy{}\PYGZhy{}force \PYGZhy{}Syu
\PYGZdl{} pacman \PYGZhy{}\PYGZhy{}force \PYGZhy{}Su
\PYGZdl{} pacman \PYGZhy{}\PYGZhy{}force \PYGZhy{}Sy base\PYGZhy{}devel msys2\PYGZhy{}devel mingw\PYGZhy{}w64\PYGZhy{}x86\PYGZus{}64\PYGZhy{}toolchain mingw\PYGZhy{}w64\PYGZhy{}x86\PYGZus{}64\PYGZhy{}qt5 mingw64/mingw\PYGZhy{}w64\PYGZhy{}x86\PYGZus{}64\PYGZhy{}hdf5 mingw64/mingw\PYGZhy{}w64\PYGZhy{}x86\PYGZus{}64\PYGZhy{}netcdf mingw64/mingw\PYGZhy{}w64\PYGZhy{}x86\PYGZus{}64\PYGZhy{}boost msys/git
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
For 32 bit

\end{itemize}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} pacman \PYGZhy{}\PYGZhy{}force \PYGZhy{}Sy
\PYGZdl{} pacman \PYGZhy{}\PYGZhy{}force \PYGZhy{}Syu
\PYGZdl{} pacman \PYGZhy{}\PYGZhy{}force \PYGZhy{}Su
\PYGZdl{} pacman \PYGZhy{}\PYGZhy{}force \PYGZhy{}Sy base\PYGZhy{}devel msys2\PYGZhy{}devel mingw\PYGZhy{}i686\PYGZhy{}toolchain mingw\PYGZhy{}i686\PYGZhy{}qt5 mingw32/mingw\PYGZhy{}i686\PYGZhy{}hdf5 mingw32/mingw\PYGZhy{}i686\PYGZhy{}netcdf mingw32/mingw\PYGZhy{}i686\PYGZhy{}boost msys/git
\end{sphinxVerbatim}
\end{quote}

\item {} 
Open mingw64.exe from the MSYS2 folder and give the following commands:
\begin{quote}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} cd \PYGZlt{}PathToInstallationFolder\PYGZgt{} \PYGZsh{}for example: cd /c/User/Admin/Desktop
\PYGZdl{} git clone https://github.com/ElucidataInc/ElMaven.git
\PYGZdl{} ./run.sh
\PYGZdl{} ./bin/El\PYGZus{}Maven\PYGZus{}0.x \PYGZsh{}for example: ./bin/El\PYGZus{}Maven\PYGZus{}0.2
\end{sphinxVerbatim}
\end{quote}

\end{itemize}

El-MAVEN loads with two windows: one for logging the application status and another for data analysis.


\subsection{Ubuntu}
\label{\detokenize{Build:ubuntu}}\begin{itemize}
\item {} 
Open the terminal and give the following commands to set up the libraries and tool chains for El-MAVEN:
\begin{quote}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} sudo apt\PYGZhy{}get update
\PYGZdl{} sudo apt\PYGZhy{}get install g++
\PYGZdl{} sudo apt\PYGZhy{}get install qt5\PYGZhy{}qmake qtbase5\PYGZhy{}dev qtscript5\PYGZhy{}dev qtdeclarative5\PYGZhy{}dev libqt5webkit5\PYGZhy{}dev libsqlite3\PYGZhy{}dev libboost\PYGZhy{}all\PYGZhy{}dev lcov libnetcdf\PYGZhy{}dev
\PYGZdl{} cd \PYGZlt{}PathToInstallationFolder\PYGZgt{} \PYGZsh{}for example: user@pc:\PYGZti{}\PYGZdl{} cd Desktop/
\PYGZdl{} git clone https://github.com/ElucidataInc/ElMaven.git
\PYGZdl{} ./run.sh
\PYGZdl{} ./bin/El\PYGZus{}Maven\PYGZus{}0.x \PYGZsh{}for example: ./bin/El\PYGZus{}Maven\PYGZus{}0.2
\end{sphinxVerbatim}
\end{quote}

\end{itemize}

El-MAVEN loads with two windows: one for logging the application status and another for data analysis.


\subsection{MacOS}
\label{\detokenize{Build:macos}}\begin{itemize}
\item {} 
Install Xcode from App store

\item {} 
Download and Install \sphinxhref{http://download.qt.io/official\_releases/qt/5.6/5.6.2/qt-opensource-mac-x64-clang-5.6.2.dmg}{Qt5.6}

\end{itemize}

This will give you the Qt5.6.2 dmg file. Using the dmg file install Qt under the directory /Users/Your\_User\_Name/
\begin{itemize}
\item {} 
Using the terminal execute the following commands:
\begin{quote}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} sudo xcodebuild \PYGZhy{}license accept
\PYGZdl{} xcode\PYGZhy{}select \PYGZhy{}\PYGZhy{}install
\PYGZdl{} /usr/bin/ruby \PYGZhy{}e \PYGZdq{}\PYGZdl{}(curl \PYGZhy{}fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)\PYGZdq{}
\PYGZdl{} brew install boost
\PYGZdl{} brew install llvm@3.7
\PYGZdl{} brew install netcdf
\PYGZdl{} cd \PYGZti{}
\PYGZdl{} touch .profile
\PYGZdl{} echo \PYGZdq{}PATH=/Users/\PYGZdl{}USER/Qt5.6.2/5.6/clang\PYGZus{}64/bin/:\PYGZdl{}PATH\PYGZdq{} \PYGZgt{} .profile
\PYGZdl{} source .profile
\PYGZdl{} mkdir \PYGZti{}/maven\PYGZus{}repo
\PYGZdl{} cd \PYGZti{}/maven\PYGZus{}repo
\PYGZdl{} git clone https://github.com/ElucidataInc/ElMaven.git
\PYGZdl{} cd ElMaven
\PYGZdl{} source \PYGZti{}/.profile
\PYGZdl{} qmake CONFIG+=debug \PYGZhy{}o Makefile build.pro
\PYGZdl{} make \PYGZhy{}j4
\end{sphinxVerbatim}
\end{quote}

\end{itemize}


\subsection{Switching Versions}
\label{\detokenize{Build:switching-versions}}
Users can switch between versions once they have compiled El-MAVEN successfully on their system.
Follow these steps to pull a specific \sphinxhref{https://elmaven.readthedocs.io/en/documentation-website/ReleaseHistory.html}{release}:
\begin{itemize}
\item {} 
Choose the version you wish to install from the list of releases. (We recommend the version tagged “Current Release”. Past releases are not stable and should be avoided)

\item {} 
Find the version tag (v0.2.x, 0.1.x, etc) on the left of release notes.

\item {} 
Open your terminal and move to the installation folder

\item {} 
Give the following commands in the terminal:
\begin{quote}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} cd ElMaven
\PYGZdl{} ./uninstall.sh
\PYGZdl{} git checkout develop
\PYGZdl{} git pull
\PYGZdl{} git checkout v0.x.y (Example: v0.4.1)
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
Build the new version using the following commands:
\begin{quote}
\begin{itemize}
\item {} 
For Windows and Ubuntu

\end{itemize}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./run.sh
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
For MacOS

\end{itemize}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} source \PYGZti{}/.profile
\PYGZdl{} qmake CONFIG+=debug \PYGZhy{}o Makefile build.pro
\PYGZdl{} make \PYGZhy{}j4
\end{sphinxVerbatim}
\end{quote}

\end{itemize}
\end{quote}

\end{itemize}


\section{User Documentation}
\label{\detokenize{Documentation:user-documentation}}\label{\detokenize{Documentation::doc}}
Welcome to the El-MAVEN user documentation!

El-MAVEN is an open source LC-MS data processing engine that is optimal for isotopomer labeling and untargeted metabolomic profiling experiments. Currently El-MAVEN exists as a desktop application that runs on Windows, Ubuntu and MacOS. The software can be used to view the mass spectra, align chromatograms, perform peak-feature detection and alignment for labeled and unlabeled mass spectrometry data. The aim of this software package is to reduce complexity of metabolomics analysis by using a highly intuitive interface for exploring and validating metabolomics data.

Find out more about the software through the links below.


\subsection{Introduction}
\label{\detokenize{Documentation:introduction}}

\subsubsection{Git Repository}
\label{\detokenize{GitRepository:git-repository}}\label{\detokenize{GitRepository::doc}}
El-MAVEN’s repository can be found on GitHub \sphinxhref{https://github.com/ElucidataInc/ElMaven}{here}.


\subsection{Getting Started}
\label{\detokenize{Documentation:getting-started}}

\subsubsection{El-MAVEN User Interface}
\label{\detokenize{IntroductiontoElMAVENUI:el-maven-user-interface}}\label{\detokenize{IntroductiontoElMAVENUI::doc}}
Following is the general workflow involved in El-MAVEN:

\noindent\sphinxincludegraphics{{Workflow}.png}

Peak detection, alignment, grouping and scoring are done multiple times for best results in the El-MAVEN workflow. Data from different cohorts can be compared using visualisation tools and easily exported to other formats.


\paragraph{El-MAVEN User Interface}
\label{\detokenize{IntroductiontoElMAVENUI:id1}}
\sphinxincludegraphics{{UI_1}.png}


\paragraph{Global Settings}
\label{\detokenize{IntroductiontoElMAVENUI:global-settings}}
Global Settings can be changed from the Options dialog \sphinxincludegraphics{{Widget_1}.png}.

\sphinxstylestrong{Instrumentation}

\sphinxincludegraphics{{UI_2}.png}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Polarity/Ionization mode}: Polarity information is required for m/z calculation. Users can set the polarity of the metabolites in their experiment from the drop-down list or set it to Auto-detect.

\item {} 
\sphinxstyleemphasis{Ionization type}: Ionization methods can affect m/z calculation. Drop-down provides a list of the most popular ionization types.

\item {} 
\sphinxstyleemphasis{Q1 accuracy}: This is the mass resolution in amu of the first quadrapole.

\item {} 
\sphinxstyleemphasis{Q3 accuracy}: This is the mass resolution in amu of the third quadrapole.

\item {} 
\sphinxstyleemphasis{Filterline}: The drop-down lists different mass ranges and allows the user to process the data in these ranges separately with different parameters. Primarily used for polarity-switching experiments.

\end{itemize}

\sphinxstylestrong{File Import}

\sphinxincludegraphics{{UI_3}.png}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Centroid Scans}: Centroid acquisition is an acquisition method where only centroid m/z and intensity are stored. Centroid m/z is calculated based on the average m/z value weighted by the intensity and m/z values are assigned based on a calibration file. Users may leave the box unchecked if they have the centroid data or check the box if centroiding has to be done in El-MAVEN.

\item {} 
\sphinxstyleemphasis{Scan Filter Polarity}: Users may choose to import scans based on the polarity of ions in the scan. Especially helpful in polarity-switching experiments.

\item {} 
\sphinxstyleemphasis{Scan Filter MS Level}: Users may choose to import only MS1 or MS2 scans. This feature can be used with MS/MS data.

\item {} 
\sphinxstyleemphasis{Scan Filter Minimum Intensity}: Sets a minimum threshold for reading in intensity values.

\item {} 
\sphinxstyleemphasis{Scan Filter Intensity Minimum Quantile Cutoff}: Scans with x\% of their intensity values below the threshold will be filtered out during import.

\item {} 
\sphinxstyleemphasis{Enable Multiprocessing}: In order to reduce the sample load time, El-MAVEN uses multiprocessing. This behavior can be changed by the user.

\end{itemize}

\sphinxstylestrong{Peak Detection}

\sphinxincludegraphics{{UI_4}.png}

\sphinxstylestrong{Peak Grouping and Grouping Settings}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{EIC Smoothing Algorithm}: Smoothing of data points helps in increasing the signal/noise ratio. There are three algorithms provided for EIC smoothing:
\begin{enumerate}
\item {} 
\sphinxstyleemphasis{Savitzky-Golay}: It preserves the original shape and features of the signal better than most other filters \sphinxhref{https://www.researchgate.net/publication/270819321\_Smoothing\_and\_Differentiation\_of\_Data\_by\_Simplified\_Least\_Squares\_Procedures}{(Learn more)}.

\item {} 
\sphinxstyleemphasis{Gaussian}: It reduces noise by averaging over the neighborhood with the central pixel having higher weight but successfully preserves sharp edges. \sphinxhref{https://people.csail.mit.edu/asolar/papers/pldi276-chaudhuri.pdf}{(Learn more)}.

\item {} 
\sphinxstyleemphasis{Moving Average}: It takes the simple average of all points over time. Signal behavior is not natural. Least preferred method for smoothing \sphinxhref{https://www.wavemetrics.com/products/igorpro/dataanalysis/signalprocessing/smoothing}{(Learn more)}.

\end{enumerate}

\item {} 
\sphinxstyleemphasis{EIC Smoothing Window}: Number of scans used for fitting in the smoothing algorithm can be adjusted here.

\item {} 
\sphinxstyleemphasis{Maximum Retention Time Difference Between Peaks}: Set a limit to retention time (RT) difference between peaks in a group. Increase the value if alignment fails to center peaks satisfactorily.

\end{itemize}

\sphinxstylestrong{Baseline Calculation}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Drop top x\% intensities from chromatogram}: Set the baseline for every peak. Baseline is obtained once x\% of the highest intensities in a peak are removed from consideration. Baseline should be set high when there is more noise in the data.

\item {} 
\sphinxstyleemphasis{Baseline Smoothing}: Number of scans used for fitting in the smoothing algorithm can be adjusted here.

\end{itemize}

\sphinxstylestrong{Peak Filtering}

\sphinxincludegraphics{{UI_5}.png}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Minimum Signal Baseline Difference}: Set the minimum difference between intensity and baseline to detect any signal as a valid peak.

\end{itemize}


\paragraph{Isotope Detection}
\label{\detokenize{IntroductiontoElMAVENUI:isotope-detection}}
\sphinxincludegraphics{{UI_6}.png}

\sphinxstylestrong{Are Samples Labeled?}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Bookmarks, peak detection, file export}: Select the labeled atoms that should be used in bookmarking, peak detection and export. D2: Deuterium, C13: Labeled carbon, N15: Labeled nitrogen, S34: Labeled sulphur.

\item {} 
\sphinxstyleemphasis{Isotopic widget}: Select the labeled atoms that should be displayed in the isotopic widget. D2: Deuterium, C13: Labeled carbon, N15: Labeled nitrogen, S34: Labeled sulphur.

\item {} 
\sphinxstyleemphasis{Number of M+n isotopes}: Set the maximum number of labeled atoms per ion in the experiment.

\item {} 
\sphinxstyleemphasis{Abundance Threshold}: Set the minimum threshold for isotopic abundance. Isotopic abundance is the ratio of intensity of isotopic peak over the parent peak.

\end{itemize}

\sphinxstylestrong{Filter Isotopic Peaks based on these criteria}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Minimum Isotope-Parent Correlation}: Set the minimum threshold for isotope-parent peak correlation. This correlation is a measure of how often they appear together.

\item {} 
\sphinxstyleemphasis{Isotope is within {[}X{]} scans of parent}: Set the maximum scan difference between isotopic and parent peaks. This is a measure of how closely they appear together on the retention time scale.

\item {} 
\sphinxstyleemphasis{Maximum \% Error to Natural Abundance}: Set the maximum natural abundance error expected. Natural abundance of an isotope is the expected ratio of amount of isotope over the amount of parent molecule in nature. Error is the difference between observed and natural abundance as a fraction of natural abundance.

\item {} 
\sphinxstyleemphasis{Correct for Natural C13 Isotope Abundance}: Check the box to correct for natural C13 abundance.

\end{itemize}

\sphinxstylestrong{EIC (XIC) {[}BETA{]}}

\sphinxincludegraphics{{UI_7}.png}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{EIC Type}: Select a method to merge EICs over m/z. There are two options:
\begin{enumerate}
\item {} 
\sphinxstyleemphasis{MAX}: Merged EIC is created by taking the maximum intensity across the m/z window at a particular scan.

\item {} 
\sphinxstyleemphasis{SUM}: Merged EIC is created by taking the sum average of intensities across the m/z window at a particular scan.

\end{enumerate}

\end{itemize}

\sphinxstylestrong{Peak Grouping}

\sphinxincludegraphics{{UI_8}.png}
\begin{quote}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Peak Grouping Score}: Peaks are assigned a grouping score to determine whether they should be grouped together. There are two formulas for grouping score calculation:
\begin{enumerate}
\item {} 
score = 1.0/((distX * A) + 0.01)/((distY * B) + 0.01) * (C * overlap)

\item {} 
score = 1.0/((distX * A) + 0.01)/((distY * B) + 0.01)

\end{enumerate}

\end{itemize}

The score depends on the following 3 parameters and their weights:
\begin{itemize}
\item {} 
\sphinxstyleemphasis{RT difference or DistX}: Difference in retention time between the peaks under comparison. Closer peaks are assigned a higher score.

\item {} 
\sphinxstyleemphasis{Intensity difference or DistY}: Difference in intensity between peaks under comparison. Smaller difference accounts for a higher score.

\item {} 
\sphinxstyleemphasis{Overlap}: Fraction of retention time overlap between the peaks under comparison. Greater overlap accounts for a higher score.
\begin{itemize}
\item {} 
Uncheck \sphinxstyleemphasis{Consider Overlap} to calculate grouping score without overlap.

\item {} 
Sliders are provided to adjust the weights attached to each of the three parameters.

\end{itemize}

\end{itemize}
\end{quote}

\sphinxstylestrong{Group Rank}

\sphinxincludegraphics{{UI_9}.png}
\begin{quote}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Group Rank Formula}: Group rank is one of the parameters for group filtering. There are two formulas below for group rank calculation:
\begin{enumerate}
\item {} 
Group Rank = ((1.1 - Q) \textasciicircum{} A) * (1/(log(I + 1)) \textasciicircum{} B)

\item {} 
Group Rank = ((1.1 - Q) \textasciicircum{} A) * (1/(log(I + 1)) \textasciicircum{} B) * (dRT) \textasciicircum{} (2 * C)

\end{enumerate}

\end{itemize}

The score depends on the following 3 parameters and their respective weights A, B and C:
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Q or Group Quality}: Maximum peak quality of a group. Peaks are assigned a quality score by a machine learning algorithm in El-MAVEN. Better quality leads to a higher rank.

\item {} 
\sphinxstyleemphasis{I or Group Intensity}: Maximum intensity of a group. Better intensity leads to a higher rank.

\item {} 
\sphinxstyleemphasis{dRT or RT difference}: Difference between expected retention time and group mean RT.
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Consider Retention Time}: Check the box to use formula (b) for group rank calculation. Formula (a) is used by default.

\item {} 
\sphinxstyleemphasis{Quality Weight}: Adjust slider to set weight for group quality in group rank calculation.

\item {} 
\sphinxstyleemphasis{Intensity Weight}: Adjust slider to set weight for group intensity in group rank calculation.

\item {} 
\sphinxstyleemphasis{dRT Weight}: Adjust slider to set weight for retention time difference in group rank calculation. The slider is disabled if Consider Retention Time is unchecked.

\end{itemize}

\end{itemize}
\end{quote}


\paragraph{Sample Upload}
\label{\detokenize{IntroductiontoElMAVENUI:sample-upload}}
\sphinxstylestrong{Load Sample Files}

Load \sphinxincludegraphics{{Widget_2}.png} sample files into El-MAVEN and click on \sphinxstyleemphasis{Show Samples Widget} \sphinxincludegraphics{{Widget_3}.png} on the widget bar to show/hide the project space. Blanks will not show up in the sample list if the file names start with ‘blan’ or ‘blank’.

\sphinxincludegraphics{{UI_10}.png}

Load sample files into El-MAVEN and click on \sphinxstyleemphasis{Show Samples Widget} on the widget bar to show/hide the project space. Blanks will not show up in the sample list if the file names start with ‘blan’ or ‘blank’.

There are three columns in the project space:
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Sample}: This column has the sample name and the random color assigned to the sample. Double-click the sample name to change the color.

\item {} 
\sphinxstyleemphasis{Set}: The set column holds the cohort name for every sample. For example: subjects and controls.

\item {} 
\sphinxstyleemphasis{Scaling}: This column holds the normalization constant for every sample. For example, all intensities in a sample will be halved if the constant is two. This is done to normalize data if sample volumes are different.

\end{itemize}

\sphinxstylestrong{Sample Space Menu}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_4}.png} \sphinxstyleemphasis{Load Project}: Sample files can be loaded here.

\item {} 
\sphinxincludegraphics{{Widget_5}.png} \sphinxstyleemphasis{Load Meta}: Users may upload a meta file with sample and set names in a comma separated file (.csv) or double-click to enter text. Meta file template is shown below:

\end{itemize}

\sphinxincludegraphics{{UI_11}.png}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_6}.png} \sphinxstyleemphasis{Save Project as}: Current state of El-MAVEN can be saved in a .mzroll file for future use. All the settings, EICs and peak tables are stored in the file and may be reloaded at any point in the future.

\item {} 
\sphinxincludegraphics{{Widget_7}.png} \sphinxstyleemphasis{Change Sample Color}: Sample colors can be changed by either clicking on this menu button or double-clicking the sample name. Users can pick a color of their choice to represent their samples.

\item {} 
\sphinxincludegraphics{{Widget_8}.png} \sphinxstyleemphasis{Remove Samples}: Apart from deselecting samples, users also have the option to remove samples from the project space. The sample files will not be deleted, only removed from El-MAVEN’s project space.

\item {} 
\sphinxincludegraphics{{Widget_9}.png} \sphinxstyleemphasis{Show/Hide Selected Samples}: Samples can be selected/deselected in batches. This is especially helpful when dealing with large datasets as the EIC window gets increasingly noisy with more samples.

\item {} 
\sphinxincludegraphics{{Widget_10}.png} \sphinxstyleemphasis{Mark Sample as Blank}: Users can select sample files and set them as blanks as depicted below. Clicking the button again will reverse the move.

\end{itemize}

\sphinxincludegraphics{{UI_12}.png}


\paragraph{Compound Database}
\label{\detokenize{IntroductiontoElMAVENUI:compound-database}}
\sphinxstylestrong{Load Reference File}

\sphinxincludegraphics{{UI_13}.png}

Reference file contains a list of metabolites and their properties that are used for peak detection. This is a comma separated (.csv) or tab separated (.tab) file with compound name, id, formula, mass, expected retention time and category. It is preferable but not necessary to have retention time information in the reference file but either mass or formula is required. In case both mass and formula are provided, formula will be used to calculate the m/z. Click on the \sphinxstyleemphasis{Show Compounds Widget} on the widget toolbar to view the compounds panel. Users may upload a new reference file or use any of the default files loaded on start-up.

\sphinxincludegraphics{{UI_14}.png}


\paragraph{EIC}
\label{\detokenize{IntroductiontoElMAVENUI:eic}}
\sphinxincludegraphics{{UI_15}.png}

An Extracted Ion Chromatogram is a graph of Intensity vs. retention time for a certain m/z range. EIC window displays the EIC for every group/compound selected or m/z range provided. The group name and/or the m/z range is displayed at the top. Following are the different menu options on top of the EIC window:
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_11}.png} \sphinxstyleemphasis{Zoom Out}: The EIC is initially zoomed-in to display the region near the expected retention time of a group. This button will zoom out and display the whole retention time range for the selected m/z range. Users may zoom in to a region by right dragging the mouse over it. Left-dragging will zoom out.

\item {} 
\sphinxincludegraphics{{Widget_12}.png} \sphinxstyleemphasis{Copy Group Information to Clipboard}: On clicking this button, group information is copied to the clipboard with every row representing a different sample.

\item {} 
\sphinxincludegraphics{{Widget_13}.png} \sphinxstyleemphasis{Bookmark as Good Group}: Users can manually curate a group as ‘good’ and store it in the bookmark table using this button. (Manual curation of groups has been covered \sphinxhref{https://github.com/ElucidataInc/ElMaven/wiki/Introduction-to-ElMaven-UI}{here})

\end{itemize}

\sphinxincludegraphics{{UI_16}.png}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_14}.png} \sphinxstyleemphasis{Bookmark as Bad Group}: User can manually curate a group as ‘bad’ and store it in the bookmark table using this button. (Manual curation of groups has been covered \sphinxhref{https://github.com/ElucidataInc/ElMaven/wiki/Introduction-to-ElMaven-UI}{here})

\end{itemize}

\sphinxincludegraphics{{UI_17}.png}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_15}.png} \sphinxstyleemphasis{History Back}: EIC window display history is recorded. Clicking this button will display the previous state of the window.

\item {} 
\sphinxincludegraphics{{Widget_16}.png} \sphinxstyleemphasis{History Forward}: EIC window display history is recorded. Clicking this button will display the next state of the window, if available.

\item {} 
\sphinxincludegraphics{{Widget_17}.png} \sphinxstyleemphasis{Save EIC Image to PDF File}: Saves the current EIC window display in a PDF file.

\item {} 
\sphinxincludegraphics{{Widget_18}.png} \sphinxstyleemphasis{Copy EIC Image to Clipboard}: Current EIC window display is copied to clipboard.

\item {} 
\sphinxincludegraphics{{Widget_19}.png} \sphinxstyleemphasis{Print EIC}: Current EIC window display can be directly printed out.

\item {} 
\sphinxincludegraphics{{Widget_20}.png} \sphinxstyleemphasis{Auto Zoom}: Auto Zoom is selected by default. It zooms-in and centers the EIC to the expected RT. The expected retention time is depicted as a dashed red line.

\end{itemize}

\sphinxincludegraphics{{UI_18}.png}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_21}.png} \sphinxstyleemphasis{Show TICs}: Displays the Total Ion Current. TIC is the sum of all intensities in a scan.

\item {} 
\sphinxincludegraphics{{Widget_22}.png} \sphinxstyleemphasis{Show Bar Plot}: Displays the peak intensity for a group in every sample. Intensity can be calculated by various methods known as quantitation types in El-MAVEN. Users can change the quantitation type from the drop-down list on the top right or choose to display other parameters like retention time and peak quality.

\end{itemize}

\sphinxincludegraphics{{UI_19}.png}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_23}.png} \sphinxstyleemphasis{Show Isotope Plot}: Displays the isotope plot for a group. Each bar in the plot represents the relative percentage of different isotopic species for the selected group in a sample.

\end{itemize}

\sphinxincludegraphics{{UI_20}.png}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_24}.png} \sphinxstyleemphasis{Show Box Plot}: Displays the boxplot for a group. The box plot shows the spread of intensities in the group and where each peak lies in relation to the median. Median of the intensities is the vertical line between the boxes.

\end{itemize}

Apart from the top menu, there are other features in the EIC window. Right-click anywhere in the window and go to Options.

\sphinxincludegraphics{{UI_21}.png}

Some of the important options are:
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Show Peaks}: Peaks are marked by the colored circles that represent the quality score of the peak. Bigger the circle, better the peak quality. This option allows the user to show/hide the peak quality score.

\item {} 
\sphinxstyleemphasis{Group Peaks Automatically}: Peak grouping happens automatically when grouping parameters are changed. To prevent automatic grouping, user can uncheck this option.

\item {} 
\sphinxstyleemphasis{Show Baseline}: Hide/Show the baseline for every peak. (Read more about baseline \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#global-settings}{here}).

\item {} 
\sphinxstyleemphasis{Show Merged EIC}: Merged EIC is the sum average of EICs across samples. It smoothens the data and helps in grouping peaks.

\item {} 
\sphinxstyleemphasis{Show EIC as Lines}: In case of large number of samples, it can get difficult to look at short individual peaks as they are obscured by larger peaks. Showing EIC as lines cleans up the display window and allows the user to look at small peaks.

\end{itemize}


\paragraph{Mass Spectra}
\label{\detokenize{IntroductiontoElMAVENUI:mass-spectra}}
Mass Spectra Widget displays each peak, its mass, and intensity for a scan. As the widget shows all detected masses in a scan, the ppm window for the EIC and consequently grouping can be adjusted accordingly. This feature is especially useful for MS/MS data and isotopic detection.

\sphinxincludegraphics{{UI_22}.png}


\paragraph{Alignment}
\label{\detokenize{IntroductiontoElMAVENUI:alignment}}
Prolonged use of the LC column can lead to a drift in retention time across samples. Alignment shifts the peak RTs in every sample to correct for this drift and brings the peaks closer to median retention time of the group.

Click on the \sphinxstyleemphasis{Align} button \sphinxincludegraphics{{Widget_25}.png} and adjust the settings.

\sphinxincludegraphics{{UI_23}.png}

The first panel in Alignment options is for Group Selection criteria. ‘Group’ here refers to a set of peaks across samples that is annotated as a particular ion.
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Group must contain at least {[}X{]} good peaks}: The value of x is set to filter out groups that do not have at least x good peaks from the alignment process. As there is only one peak per sample for a group, this value should not exceed the number of samples in your project. This option allows the users to discard groups with very few good peaks under the assumption that those could be stray peaks.

\item {} 
\sphinxstyleemphasis{Limit total number of groups in alignment to}: Users can change the number of groups being used for alignment in case there are too many groups detected after the peak detection process.

\item {} 
\sphinxstyleemphasis{Peak Grouping Window}: This value controls the number of scans required to get the most accurate peaks. Enter a high number if the reproducibility is low to ensure best results.

\end{itemize}

The next panel is for \sphinxstyleemphasis{Peak Selection} settings:
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Minimum Peak Intensity}: The intensity value can be adjusted to only look at high or low intensity peaks in case you have prior information about the concentration of metabolite users are looking for.

\item {} 
\sphinxstyleemphasis{Minimum peak S/N ratio}: This is the minimum signal to noise ratio of your experiment. Increase the value if you see too much noise in the data.

\item {} 
\sphinxstyleemphasis{Minimum Peak Width}: This is the least number of scans to be considered to evaluate the width of any peak.

\item {} 
\sphinxstyleemphasis{Peak Detection Algorithm}: Select the \sphinxstyleemphasis{Compound Database Search} algorithm and then choose an appropriate database from the next drop-down menu.

\end{itemize}

The \sphinxstyleemphasis{Alignment Algorithm} panel provides the following options:
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Alignment Algorithm}: There are three alignment algorithms available in El-MAVEN: Obi-Warp, Poly fit and Loess fit. Loess fit has been released as a beta feature for now.

\item {} 
\sphinxstyleemphasis{Maximum number of Iterations}: This parameter is only required for Poly fit algorithm. Enter the number of times El-MAVEN should fit a model to the data in order to align it.

\item {} 
\sphinxstyleemphasis{Polynomial Degree}: This is the degree of the non-linear model we are trying to fit. Recommended settings are entered by default.

\end{itemize}

Click on \sphinxstyleemphasis{Align} at the bottom.

\sphinxstylestrong{Alignment Visualizations}

El-MAVEN provides three visualizations for alignment analysis.
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Show Alignment Visualization}: Click on \sphinxincludegraphics{{Widget_26}.png} in the widget bar to open this visualization. Click on any grouped peak to look at its delta RT vs RT graph as shown.

\end{itemize}

\sphinxincludegraphics{{UI_24}.png}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Show Alignment Visualization (For All Groups)}: Click on \sphinxincludegraphics{{Widget_27}.png} in the widget bar for this visualization.

\end{itemize}

\sphinxincludegraphics{{UI_25}.png}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Show Alignment Polynomial Fit}: Click on \sphinxincludegraphics{{Widget_28}.png} in the widget bar for Poly fit alignment.

\end{itemize}

\sphinxincludegraphics{{UI_26}.png}

The above graphs give a clear indication of how aligned/misaligned the peaks are. Users may run alignment again with different parameters if required (or with a different algorithm).


\paragraph{Peak Detection}
\label{\detokenize{IntroductiontoElMAVENUI:peak-detection}}
Peak detection algorithm pulls the EICs, detects peaks and performs grouping and filtering based on parameters controlled by the users. The algorithm groups identical peaks across samples and calculates the quality score by a machine learning algorithm. Click on the \sphinxstyleemphasis{Peaks} icon \sphinxincludegraphics{{Widget_29}.png} on the top to open the settings dialog.

There are 3 tabs for setting Peak Detection parameters:

\sphinxstylestrong{1. Feature Detection Selection}

\sphinxincludegraphics{{UI_27}.png}

The Feature Detection Selection panel has the following parameters:
\begin{quote}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Automated Feature Detection}: This is one of the two strategies for finding peaks. Automated search creates thousands of mass slices across the whole m/z and retention time space to find all peaks present in the sample. This strategy is used when looking for new/unknown metabolites in the samples.
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Mass Domain Resolution}: This value defines the m/z range of every mass slice in parts per million

\item {} 
\sphinxstyleemphasis{Time Domain Resolution}: This value defines the scan range (or retention time range) of every mass slice

\item {} 
\sphinxstyleemphasis{Limit Mass Range}: User can limit the automated search to a range of m/z according to their requirements

\item {} 
\sphinxstyleemphasis{Limit Time Range}: User can limit the automated search to a retention time range according to their requirements

\end{itemize}

\item {} 
\sphinxstyleemphasis{Compound Database Search}: Database search is used to search for compounds listed in the reference file using their m/z information. For better accuracy, retention time information can also be used for the search.
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Select Database}: Select a desired reference file for the search from the drop-down list

\item {} 
\sphinxstyleemphasis{EIC Extraction Window}: Provide a ppm buffer range to all compound masses. A larger window is useful for processing low resolution data. The window should be smaller for high resolution data to reduce noise.

\item {} 
\sphinxstyleemphasis{Match Retention Time}: Enable/disable use of retention time information along with m/z to perform database search. Compounds can have different retention times in every experiment, therefore this option should only be checked if the reference file is specific to the experiment and the sampled used. Enter the time buffer in the accompanying box.

\item {} 
\sphinxstyleemphasis{Limit Number of Reported Groups Per Compound}: Multiple groups can be annotated as the same compound, especially when retention time is not taken into consideration for the search. Users can set the value to only report X best groups according to their rank. The group rank formula will be discussed later in the tutorial.

\end{itemize}

\item {} 
\sphinxstyleemphasis{Match Fragmentation}: This panel is activated for MS/MS data.

\item {} 
\sphinxstyleemphasis{Report Isotopic Peaks}: Check this box to find and report isotopic peaks for labeled data.

\end{itemize}

To perform peak detection with reference, check the box next to \sphinxstyleemphasis{Compound Database Search} and choose the appropriate database. The \sphinxstyleemphasis{EIC Extraction Window} should be set according to the instrument’s resolving power. Select the \sphinxstyleemphasis{Match Retention Time} option if you wish to search for compounds using both the m/z ratio and retention time value given in the database. In case of a generic database, searching by retention time is not recommended.
\end{quote}

\sphinxstylestrong{2. Group Filtering}

\sphinxincludegraphics{{UI_28}.png}

After grouping is done, groups that do not fulfill the criteria shown above are filtered out.
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Minimum Peak Intensity}: Groups with no peak intensities above this threshold are filtered out. The drop-down list beside the input box defines how intensity is calculated. Different methods of intensity calculation are known as quantitation types. The slider below can be adjusted to change the minimum percentage of peaks per group that must pass the threshold (minimum number of peaks is 1).

\item {} 
\sphinxstyleemphasis{Minimum Quality}: Quality of peaks is calculated using a machine learning algorithm. Groups with no peak qualities above this threshold are filtered out. The slider below can be adjusted to change the minimum percentage of peaks per group that must pass the threshold (minimum number of peaks is 1).

\item {} 
\sphinxstyleemphasis{Minimum Signal/Blank Ratio}: Signal/Blank ratio is the ratio of peak intensity over maximum intensity observed in blanks. Groups with no peaks above this threshold are filtered out. The slider can be adjusted to change the minimum percentage of peaks per group that must pass the threshold (minimum number of peaks is 1). This helps in filtering out peaks that are also present in blanks.

\item {} 
\sphinxstyleemphasis{Minimum Signal/Baseline Ratio}: Signal/Baseline ratio is the ratio of peak intensity over baseline value for that peak. Baseline calculation is used to filter out noise in the signal and will be discussed later in the tutorial. The slider can be adjusted to change the minimum percentage of peaks per group that must pass the threshold (minimum number of peaks is 1).

\item {} 
\sphinxstyleemphasis{Minimum Peak Width}: Peak width is equal to the number of scans that a peak is spread over. Groups with no peak widths above this threshold are filtered out. Spurious signals can be filtered out using this option.

\item {} 
\sphinxstyleemphasis{Peak Classifier Model File}: This is the default model that is used by the machine learning algorithm for classifying peaks according to their quality.

\end{itemize}

Change the settings according to the data and click on \sphinxstyleemphasis{Find Peaks} to run peak detection. For beginners, performing peak detection with default values at first is recommended. Users may then adjust the settings depending on their results.

\sphinxstylestrong{3. Method Summary}

\sphinxincludegraphics{{UI_29}.png}


\paragraph{Peak Table}
\label{\detokenize{IntroductiontoElMAVENUI:peak-table}}
\sphinxincludegraphics{{UI_30}.png}

Groups information obtained after Peak Detection is stored and displayed in the form of a Peak Table with a row representing a group and its corresponding features in columns. Users can show/hide the peak table by clicking on \sphinxincludegraphics{{Widget_29}.png} the widget bar.

\sphinxstylestrong{Peak Table Features}

Following are the different features/columns in a peak table:
\begin{itemize}
\item {} 
\sphinxstyleemphasis{\#}: is the serial number for a group

\item {} 
\sphinxstyleemphasis{ID}: Group ID is assigned according to the search mode used during peak detection. In case of Automated search, groups are named by their m/z and retention time values separated by ‘@’ sign. For example, ID for a group with 230.2 m/z and 1.89 RT will be given as \sphinxhref{mailto:'230.2@1.89}{‘230.2@1.89}’. In case of Database search, groups are annotated as a compound from the reference file. For example, ‘malate’.

\item {} 
\sphinxstyleemphasis{Observed m/z}: is the median m/z of the group.

\item {} 
\sphinxstyleemphasis{Expected m/z}: is the m/z value provided in the reference file for the compound represented by the group. This field is populated only in case of Database search.

\item {} 
\sphinxstyleemphasis{rt}: is the median retention time of the group.

\item {} 
\sphinxstyleemphasis{rt delta}: is the difference between expected retention time from the reference file and the observed RT. This field is set to -1 in case of Automated Search.

\item {} 
\sphinxstyleemphasis{\#peaks}: is the number of peaks in the group.

\item {} 
\sphinxstyleemphasis{\#good}: is the number of good peaks in a group. A good peak is defined as one with its quality score above the defined threshold in Peak Detection dialog.

\item {} 
\sphinxstyleemphasis{Max Width}: is the maximum peak width in a group. Peak width is defined as the number of scans over which a peak is spread.

\item {} 
\sphinxstyleemphasis{Max AreaTop}: is the maximum peak AreaTop intensity in a group. AreaTop is one of the quantitation types used to represent peak intensity in El-MAVEN. Read more about the different quantitation types \sphinxhref{https://github.com/ElucidataInc/ElMaven/wiki/Introduction-to-ElMaven-UI}{here}.

\item {} 
\sphinxstyleemphasis{Max S/N}: is the maximum peak signal/noise ratio in a group.

\item {} 
\sphinxstyleemphasis{Max Quality}: is the maximum peak quality score in a group.

\item {} 
\sphinxstyleemphasis{Rank}: is the group rank. The formula and parameters involved have been explained \sphinxhref{https://github.com/ElucidataInc/ElMaven/wiki/Introduction-to-ElMaven-UI}{here}.

\end{itemize}

\sphinxstylestrong{Peak Table Menu Bar}

\sphinxincludegraphics{{UI_31}.png}

Multiple groups can be annotated as the same compound especially when retention time information is not used during Database search. The peak table provides options for filtering, comparing or exporting data from the table. Following are the different menu options available in the peak table:
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_30}.png} \sphinxstyleemphasis{Switch between group and peak views}: Switching to Peak view displays only Peak information. This includes group number, group ID, Expected m/z, Observed m/z, retention time and intensity of all peaks in the group with sample names as the respective column headers. Peak intensity cells are colored based on their relative values in a group. Highest intensity value has the lightest color and vice-versa.

\end{itemize}

\sphinxincludegraphics{{UI_32}.png}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_13}.png} \sphinxstyleemphasis{Mark Group as Good}: Used to manually curate selected peaks as ‘good’. User can also press ‘G’ on their keyboard for the same. Manual curation has been described \sphinxhref{https://github.com/ElucidataInc/ElMaven/wiki/Introduction-to-ElMaven-UI}{here}.

\end{itemize}

\sphinxincludegraphics{{UI_33}.png}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_14}.png} \sphinxstyleemphasis{Mark Group as Bad}: Used to manually reject peaks by marking them as ‘bad’. User can also press ‘B’ on their keyboard for the same. Manual curation has been described \sphinxhref{https://github.com/ElucidataInc/ElMaven/wiki/Introduction-to-ElMaven-UI}{here}.

\end{itemize}

\sphinxincludegraphics{{UI_34}.png}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_31}.png} \sphinxstyleemphasis{Train Neural Net}: Used to retrain the neural net algorithm to recognize good/bad peaks. User manually curates 100 peaks to train the algorithm.

\item {} 
\sphinxincludegraphics{{Widget_8}.png} \sphinxstyleemphasis{Delete Group}: Deletes the selected group(s) from the peak table.

\item {} 
\sphinxincludegraphics{{Widget_32}.png} \sphinxstyleemphasis{Show Scatter Plot}: Opens the Scatter plot widget used to compare different cohorts via Scatter plot and Volcano plot.

\end{itemize}

The remaining are export options and will be detailed in the \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#id10}{Export} section.


\paragraph{Statistics}
\label{\detokenize{IntroductiontoElMAVENUI:statistics}}
El-MAVEN comes equipped with a statistics module for comparing data across different cohorts. Users can set the sample cohorts either by editing the Set column in the Sample space, or upload a meta file with sample and cohort names as detailed above under the \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#sample-upload}{Sample Space Menu} section.

The statistics module can be accessed through the Peak Table menu.

\sphinxincludegraphics{{UI_35}.png}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Set1/Set2}: Select two cohorts to be compared

\item {} 
\sphinxstyleemphasis{Min Log2 Fold Difference}: Fold difference is a measure of how much the intensity of a group changes from one cohort to another. User can set the minimum threshold for this value in log$_{\text{2}}$format.

\item {} 
\sphinxstyleemphasis{Min Intensity}: Groups with all peak intensities less than this value will be filtered out from the comparison process.

\item {} 
\sphinxstyleemphasis{p value}: A t-test is performed to find if the intensity distributions of the two selected cohorts are significantly different from each other. This test returns a p-value indicating how significantly different a group behaves between the two cohorts. A lower p-value shows higher significance.

\item {} 
\sphinxstyleemphasis{Set Missing Values}: User can set the default intensity value to be used in case the group is missing from a sample.

\item {} 
\sphinxstyleemphasis{Min. Good Sample}: Groups should have a minimum number of good peaks (based on peak quality score) to be considered for comparison.

\item {} 
\sphinxstyleemphasis{FDR Correction}: False discovery rate is the expected proportion of false positives in a test. There are a number of ways to correct for false positives. (\sphinxhref{http://nebc.nerc.ac.uk/courses/GeneSpring/GS\_Mar2006/Multiple\%20testing\%20corrections.pdf}{Read More})

\item {} 
\sphinxstyleemphasis{Compare Sets}: Click to get comparison results.

\end{itemize}

\sphinxstyleemphasis{Compare Sets} opens the scatter plot by default

\sphinxincludegraphics{{UI_36}.png}
\begin{itemize}
\item {} 
\sphinxincludegraphics{{Widget_11}.png} \sphinxstyleemphasis{Zoom Out}: Zooms out of the plot.

\item {} 
\sphinxincludegraphics{{Widget_33}.png} \sphinxstyleemphasis{Compare Samples}: Opens the compare samples dialog again to adjust settings.

\item {} 
\sphinxincludegraphics{{Widget_32}.png} \sphinxstyleemphasis{Scatter Plot}: The axes represent the average peak intensity (Peak Height) for sample 1 and 2 respectively. Each bubble is a group. The bubble size represents fold change between the samples. The significance (or p-value) of the fold change is represented by the bubble color. Red and blue signify higher intensity in sample 1 and 2 respectively. Opaqueness of the bubble represents the significance (or inverse of p-value) of the fold change between cohorts.

\item {} 
\sphinxincludegraphics{{Widget_34}.png} \sphinxstyleemphasis{Volcano Plot}: The axes represent fold change and significance of fold change respectively. Red and blue bubbles represent positive and negative fold change respectively.

\item {} 
\sphinxincludegraphics{{Widget_8}.png} \sphinxstyleemphasis{Delete}: Deletes a data point from the graph.

\item {} 
\sphinxincludegraphics{{Widget_29}.png} \sphinxstyleemphasis{Scatter Plot Table}: A separate Peak Table is created with all filtered groups being used for statistical analysis. User can also export these in a CSV or JSON.

\end{itemize}


\paragraph{Export}
\label{\detokenize{IntroductiontoElMAVENUI:id10}}
Users can either save the state of the project or export only relevant data from the peak table. These are the different export options available in El-MAVEN:

\sphinxincludegraphics{{UI_37}.png}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Save Project as}: This option is available in the File menu. It saves all peak tables and current settings in a .mzroll file. On loading the .mzroll file, all sample files are uploaded and the peak tables and EIC are available. If the user wishes to save only certain Peak Tables, they can click on \sphinxincludegraphics{{Widget_6}.png} at the top of the Peak Table(s). This will only store that specific peak table instead of all.

\item {} 
\sphinxstyleemphasis{Generate PDF Report}: This option is available on \sphinxincludegraphics{{Widget_35}.png} at the top of the Peak Table. It saves all EICs with their corresponding bar plots in a PDF file.

\item {} 
\sphinxstyleemphasis{Export Groups to SpreadSheet (.csv)}: This option is available on top of the Peak Table \sphinxincludegraphics{{Widget_36}.png}. You can choose to export the whole table or a subset of the data. There are 4 possible selections: export only selected groups, export all groups, export only good groups or export only bad groups. The data is stored in a comma separated file.

\end{itemize}

\sphinxincludegraphics{{UI_38}.png}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Export EICs to Json}: This option is available on top of the Peak Table \sphinxincludegraphics{{Widget_37}.png}. It exports all EICs to a Json file.

\end{itemize}


\subsubsection{El-MAVEN Command Line Interface}
\label{\detokenize{IntroductiontoElMAVENCLI:el-maven-command-line-interface}}\label{\detokenize{IntroductiontoElMAVENCLI::doc}}
The peakdetector executable in El-MAVEN can load sample files, perform peak detection and save the resulting peaks in a CSV or mzroll file. It reads a configuration file with user-set parameters and paths to input and output files.

This executable is not part of installer versions of El-MAVEN. Please clone the repository and build \sphinxhref{https://github.com/ElucidataInc/ElMaven/blob/develop/README.md\#compilation}{El-MAVEN} to use this feature.


\paragraph{Create a default configuration file}
\label{\detokenize{IntroductiontoElMAVENCLI:create-a-default-configuration-file}}
Users need an XML file to specify their configuration and specify directory paths. The XML template can be generated using a simple command and modified according to user requirements.
\begin{quote}
\begin{itemize}
\item {} 
Open the terminal

\item {} 
Move to your local El-MAVEN folder:

\end{itemize}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} cd user@pc:\PYGZti{}/ElMaven/ElMaven
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
Execute the following command
\begin{quote}
\begin{itemize}
\item {} 
For Windows and Ubuntu:

\end{itemize}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./bin/peakdetector.exe \PYGZhy{}\PYGZhy{}defaultxml
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
For MacOS:

\end{itemize}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./bin/peakdetector.app/Contents/MacOS/peakdetector \PYGZhy{}\PYGZhy{}xml config.xml
\end{sphinxVerbatim}
\end{quote}

\item {} 
A default XML will be created in your current folder

\end{itemize}
\end{quote}


\paragraph{Peak detector parameters}
\label{\detokenize{IntroductiontoElMAVENCLI:peak-detector-parameters}}
CLI provides a limited number of parameters for user-modification. Default values are used for all other parameters used in El-MAVEN GUI.

The template XML contains three sets of parameters:

\sphinxstylestrong{1. OptionalDialogArguments: Global parameters}
\begin{itemize}
\item {} 
Users need to know the ionization mode and accepted values are 0 (neutral), -1 (negative) and +1 (positive): \sphinxcode{\sphinxupquote{ionizationMode type="int" value="-1"}}

\item {} 
Ionization charge. This is the number of positive or negative charge added to every ion. Accepted values are positive integers: \sphinxcode{\sphinxupquote{charge type="int" value="1"}}

\item {} 
Mass accuracy. Set the +/- ppm range for m/z values: \sphinxcode{\sphinxupquote{compoundPPMWindow type="float" value="5"}}

\end{itemize}

\sphinxstylestrong{2. PeaksDialogArguments: Peak detection parameters}
\begin{quote}
\begin{itemize}
\item {} 
Peak Detection Algorithm. Accepted values: “0” for Compound Database Search, “1” (or higher) for Automated Feature Detection: \sphinxcode{\sphinxupquote{processAllSlices type="int" value="0"}}

\end{itemize}

Following are the arguments used in Automated Peak Detection and their GUI equivalents. These can be found in the ‘Feature Detection Selection’ tab of Peaks Dialog:
\begin{itemize}
\item {} 
‘Limit m/z Range’ minimum: \sphinxcode{\sphinxupquote{minScanMz type="float" value="0"}}

\item {} 
‘Limit m/z Range’ maximum: \sphinxcode{\sphinxupquote{maxScanMz type="float" value="1e+009"}}

\item {} 
‘Limit Time Range’ minimum: \sphinxcode{\sphinxupquote{minScanRt type="float" value="0"}}

\item {} 
‘Limit Time Range’ maximum: \sphinxcode{\sphinxupquote{maxScanRt type="float" value="1e+009"}}

\item {} 
‘Limit Intensity Range’ minimum: \sphinxcode{\sphinxupquote{minScanIntensity type="float" value="50000"}}

\item {} 
‘Limit Intensity Range’ maximum: \sphinxcode{\sphinxupquote{maxScanIntensity type="float" value="1e+010"}}

\item {} 
‘Mass Domain Resolution’: \sphinxcode{\sphinxupquote{ppmMerge type="float" value="20"}}

\end{itemize}

Following are the arguments used in Compound database search and their GUI equivalents. These can be found in the ‘Feature Detection Selection’ tab of Peaks Dialog:
\begin{itemize}
\item {} 
‘Select Database’. Enter full path to the database file: \sphinxcode{\sphinxupquote{Db type="string" value="0"}}

\item {} 
‘Match Retention Time (+/-)’. Value of ‘0’ will set the flag as false. Positive integer value will set the allowed deviation in minutes: \sphinxcode{\sphinxupquote{matchRtFlag type="int" value="0"}}

\item {} 
‘Limit Number of Reported Groups per Compound’:  \sphinxcode{\sphinxupquote{eicMaxGroups type="int" value="0"}}

\end{itemize}

Isotope detection. Following are the settings used to pull isotopes in case of database search and their GUI equivalents. These can be found in the ‘Isotope Detection Options’ in the Options Dialog:
\begin{itemize}
\item {} 
Choose labels present in your data. Enter a four digit value of ‘1’s and ‘0’s each representing a label: D2, N15, S34 and C13. For eg. 0000 signifies no labels, 0001 signifies your data has C13 label and 1111 signifies your data has all four labels: \sphinxcode{\sphinxupquote{pullIsotopes type="int" value="0"}}

\end{itemize}

EIC processing. Following are the parameters used for EIC processing and filtering and their GUI equivalents. These can be found under the ‘Peak Detection’ tab of Options Dialog:
\begin{itemize}
\item {} 
‘Maximum Retention Time Difference between Peaks’:  \sphinxcode{\sphinxupquote{grouping \_maxRtWindow type="float" value="0.5"}}

\item {} 
‘EIC smoothing window’: \sphinxcode{\sphinxupquote{eicSmoothingWindow type="int" value="10"}}

\end{itemize}

Group Filtering. Following are the parameters used for filtering detected groups and their GUI equivalents. These can be found under the ‘Group Filtering’ tab of Peak Dialog:
\begin{itemize}
\item {} 
‘Minimum Peak Intensity’: \sphinxcode{\sphinxupquote{minGroupIntensity type="float" value="5000"}}

\item {} 
Quantitation type of the intensity set in ‘Minimum Peak Intensity’. Accepted values are- “0”:AreaTop, “1”:Area, “2”:Height, “3”:AreaNotCorrected: \sphinxcode{\sphinxupquote{quantitationType type="int" value="0"}}

\item {} 
‘At least x\% peaks above minimum intensity’ slider. Enter value in percentage. 0\% will default to 1 peak: \sphinxcode{\sphinxupquote{quantileIntensity type="float" value="0"}}

\item {} 
‘Minimum Quality’: \sphinxcode{\sphinxupquote{minQuality type="float" value="0.5"}}

\item {} 
‘At least x\% peaks above minimum quality’ slider: \sphinxcode{\sphinxupquote{quantileQuality type="float" value="0"}}

\item {} 
‘Minimum peak width’. Enter number of scans: \sphinxcode{\sphinxupquote{minPeakWidth type="int" value="1"}}

\item {} 
‘Minimum Signal/Baseline Ratio’: \sphinxcode{\sphinxupquote{minSignalBaseLineRatio type="float" value="2"}}

\item {} 
‘Minimum Good Peak/Group’. Value should be less or equal to the number of samples: \sphinxcode{\sphinxupquote{minGoodGroupCount type="int" value="1"}}

\item {} 
‘Peak Classifier Model File’. The default model file can be found in the bin folder of El-MAVEN installation. Enter full path to the ‘default.model’ file: \sphinxcode{\sphinxupquote{model type="string" value="0"}}

\item {} 
Not used by CLI: \sphinxcode{\sphinxupquote{rtStepSize type="float" value="20"}}

\end{itemize}
\end{quote}

\sphinxstylestrong{3. GeneralArguments: General parameters}
\begin{itemize}
\item {} 
Alignment. Positive integer value will run an alignment algorithm: \sphinxcode{\sphinxupquote{alignSamples type="int" value="0"}}

\item {} 
Export EIC in JSON. Positive integer value will save EICs JSON in user-specifed output folder: \sphinxcode{\sphinxupquote{saveEicJson type="int" value="0"}}

\item {} 
Output directory. Enter full path to desired output folder: \sphinxcode{\sphinxupquote{outputdir type="string" value="0"}}

\item {} 
Save Project. Positive integer value will save the project as a .mzroll file in user-specified output folder. This file can be loaded in El-MAVEN GUI for further processing or visualization: \sphinxcode{\sphinxupquote{savemzroll type="int" value="0"}}

\item {} 
Sample Path. Enter full path to a sample file in each row: \sphinxcode{\sphinxupquote{samples type="string" value="0"}}

\end{itemize}


\paragraph{Run}
\label{\detokenize{IntroductiontoElMAVENCLI:run}}
Once the parameters and directory paths have been set in the configuration file, run peak detection from the terminal using the following command from the El-MAVEN root directory:
\begin{quote}
\begin{itemize}
\item {} 
Windows and Ubuntu:

\end{itemize}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./bin/peakdetector.exe \PYGZhy{}\PYGZhy{}xml config.xml
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
MacOS:

\end{itemize}

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ./bin/peakdetector.app/Contents/MacOS/peakdetector \PYGZhy{}\PYGZhy{}xml config.xml
\end{sphinxVerbatim}
\end{quote}

The resulting CSV file (and other files depending on the configuration) can be found in the specified output directory.


\paragraph{Help}
\label{\detokenize{IntroductiontoElMAVENCLI:help}}
To print the help commands, execute the following:

\fvset{hllines={, ,}}%
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} Peakdetector.exe \PYGZhy{}h
\end{sphinxVerbatim}


\subsection{Tutorials}
\label{\detokenize{Documentation:tutorials}}

\subsubsection{Unlabeled LC-MS Workflow}
\label{\detokenize{UnlabeledLCMSWorkflow:unlabeled-lc-ms-workflow}}\label{\detokenize{UnlabeledLCMSWorkflow::doc}}
This is a tutorial for processing Unlabeled LC/MS data files through El-MAVEN.


\paragraph{Preprocessing}
\label{\detokenize{UnlabeledLCMSWorkflow:preprocessing}}
msConvert is a command-line/GUI tool that is used to convert between various mass spectroscopy data formats, developed and maintained by proteoWizard. Raw data files obtained from mass spectrometers need to be converted to certain acceptable formats before processing in El-MAVEN.

\sphinxstylestrong{Input}

msConvert supports the following formats:
\begin{itemize}
\item {} 
.mzXML

\item {} 
.mzML

\item {} 
.RAW ThermoFisher

\item {} 
.RAW Walters

\item {} 
.d Agilent

\item {} 
.wiff ABSciex

\end{itemize}

The settings used for msConvert as a GUI tool are captured in the following screenshots:

\sphinxincludegraphics{{ULCMS_1}.png}

\begin{sphinxadmonition}{note}{Note:}
zlib compression is enabled by default in msConvert. El-MAVEN in its current form does not support zlib compression. It is important to uncheck “Use zlib compression” box.
\end{sphinxadmonition}

\sphinxstylestrong{Output}

msConvert can convert to an array of different formats but El-MAVEN primarily uses .mzXML and .mzML formats.


\paragraph{Launch El-MAVEN}
\label{\detokenize{UnlabeledLCMSWorkflow:launch-el-maven}}
Once sample files are ready for processing, launch El-MAVEN.

\sphinxincludegraphics{{ULCMS_2}.png}


\paragraph{Adjust Global Settings}
\label{\detokenize{UnlabeledLCMSWorkflow:adjust-global-settings}}
Global Settings can be changed from the \sphinxstyleemphasis{Options} dialog \sphinxincludegraphics{{Widget_1}.png}. There are 9 tabs in the dialog. Each of these tabs has parameters related to a different module in El-MAVEN. For example, a tab for Instrumentation details, a tab for File Import settings etc.

\sphinxincludegraphics{{ULCMS_3}.png}

To know more about the functionality of different tabs and their settings, users can see the \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#global-settings}{Widgets page}. Please be sure to set the desired settings before processing an input file.


\paragraph{Load Samples}
\label{\detokenize{UnlabeledLCMSWorkflow:load-samples}}
Users can go to File in the Menu and click on \sphinxstyleemphasis{Load Samples\textbar{}Projects\textbar{}Peaks} option. Then navigate to the folder containing the sample data, select all .mzXML or .mzML files and click Open. A loading bar displays the progress at the bottom.

\sphinxincludegraphics{{ULCMS_4}.png}

When the samples have loaded, users should see a sample panel on the left side. If it is not displayed automatically, click on the \sphinxstyleemphasis{Show Samples Widget} button on the toolbar. El-MAVEN automatically assigns a color to every sample. Users can select/deselect any sample by clicking the checkbox on the left of the sample name.

\sphinxincludegraphics{{ULCMS_5}.png}


\paragraph{Load Compound Database}
\label{\detokenize{UnlabeledLCMSWorkflow:load-compound-database}}
Users can click on \sphinxstyleemphasis{Compounds} option in the leftmost menu, navigate to the folder containing the standard database file, select the appropriate .csv file and click \sphinxstyleemphasis{Open}. Alternatively, users may use any of the default files loaded on start-up.

\sphinxincludegraphics{{ULCMS_6}.png}

This is a sample Compound Database:

\sphinxincludegraphics{{ULCMS_7}.png}

It lists all metabolite names, chemical formula, HMDB ID, and the class/category of compounds they belong to (if known).


\paragraph{Mark Blanks}
\label{\detokenize{UnlabeledLCMSWorkflow:mark-blanks}}
Users can mark the blanks by selecting the blank samples from the list on screen, and clicking on the \sphinxstyleemphasis{Set as a Blank Sample} icon \sphinxincludegraphics{{Widget_10}.png} in Samples menu.

\sphinxincludegraphics{{ULCMS_8}.png}

Multiple blanks can be marked together. The blanks will appear black as shown in the image below:

\sphinxincludegraphics{{ULCMS_9}.png}


\paragraph{Alignment}
\label{\detokenize{UnlabeledLCMSWorkflow:alignment}}
Prolonged use of the LC column can lead to a drift in retention time across samples. Alignment shifts the peak retention time in every sample to correct for this drift and brings the peaks closer to median retention time of the group.

\sphinxincludegraphics{{ULCMS_10}.png}

In the above image, EIC for a UTP group is displayed. If the samples were aligned, all peaks would lie at the same retention time. Since this is not the case, the samples need to be aligned.

\sphinxstyleemphasis{Alignment visualization} \sphinxincludegraphics{{Widget_26}.png} can be used to judge the extent of deviation from median retention time.

\sphinxincludegraphics{{ULCMS_11}.png}

In the above visualization, each box represents a peak from the selected group at its current retention time. Samples are said to be perfectly aligned when all peak boxes lie on the same vertical axis. The peaks are considerably scattered in the above image and therefore the samples should be aligned for better grouping of peaks.

\sphinxstylestrong{Perform Alignment}

\sphinxstyleemphasis{Alignment settings} can be adjusted using the Align button \sphinxincludegraphics{{Widget_25}.png}. This example was aligned with \sphinxstyleemphasis{Poly fit} algorithm with default parameters.

\sphinxincludegraphics{{ULCMS_12}.png}

Post-alignment the peaks in the group should appear closer to the median retention time of the group.

\sphinxincludegraphics{{ULCMS_13}.png}

\sphinxincludegraphics{{ULCMS_14}.png}

Pre-alignment, the peaks were considerably scattered while the aligned peaks lie nearly on the same axis. Users can run alignment again with different parameters if required (or with a different algorithm). Further details on Alignment settings are available on the \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#alignment}{Widgets page}.


\paragraph{Peak Grouping}
\label{\detokenize{UnlabeledLCMSWorkflow:peak-grouping}}
Peak grouping is an integral part of the El-MAVEN workflow that categorizes all detected peaks into groups on the basis of certain user-controlled parameters. A group score is calculated for every peak during the process. The formula for this score takes into account the difference in retention time, intensities between peaks (smaller difference leads to a better score) and any existing overlap between them (higher extent of overlap leads to better score). All three parameters have certain weights attached to them that can be controlled by the users. The formula for the score is shown in the image. More details on it can be found on the \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#global-settings}{Widgets page}.

\sphinxincludegraphics{{ULCMS_15}.png}

\sphinxincludegraphics{{ULCMS_16}.png}

The above image shows two groups in the EIC window. The highlighted (solid circles) peaks belong to group A, the peaks to its left with empty circles belong to another group B. The short peaks in group A that are close to the baseline and peaks in group B come from the same samples. Additionally, the high intensity peaks of group A have a similar peak shape to group B peaks. These peaks might have been wrongly classified into separate groups because of the difference in retention time range of the two sets of peaks. The weights attached to difference in retention time and intensities, and extent of overlap can be adjusted for better grouping.

Grouping parameters can be changed from the Options dialog \sphinxincludegraphics{{Widget_1}.png}.

\sphinxincludegraphics{{ULCMS_17}.png}

\sphinxincludegraphics{{ULCMS_18}.png}

Giving less priority to difference in retention time and intensities results in the two groups being merged into a single  group while the peaks that lay close to the baseline are no longer classified as valid peaks.


\paragraph{Baseline}
\label{\detokenize{UnlabeledLCMSWorkflow:baseline}}
When measuring a number of peaks, it is often more effective to subtract an estimated baseline from the data. This baseline should be set where ideally no peaks occur. Although sometimes the program sets a particular baseline such that one or more peaks occur below that baseline value. In the following image, the dashed line represents each baseline:

\sphinxincludegraphics{{ULCMS_19}.png}

The corresponding peaks are indicated with solid circles:

\sphinxincludegraphics{{ULCMS_20}.png}

The baseline correction can be done in the \sphinxstyleemphasis{Peak Detection} tab by clicking on \sphinxstyleemphasis{Options} button:

\sphinxincludegraphics{{ULCMS_21}.png}

Further details on settings can be accessed \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#peak-detection}{here}.

The \sphinxstyleemphasis{m/z} option scans the groups to find any specific m/z value and plot its corresponding EIC. The +/- option to its right is to specify the expected mass resolution error in parts per million (ppm).

\sphinxincludegraphics{{ULCMS_23}.png}


\paragraph{Mass Spectra}
\label{\detokenize{UnlabeledLCMSWorkflow:mass-spectra}}
Mass Spectra Widget \sphinxincludegraphics{{Widget_38}.png} displays each peak, its mass, and intensity for a scan. As the widget shows all detected masses in a scan, the ppm window for the EIC and consequently grouping can be adjusted accordingly. This feature is especially useful for MS/MS data and isotopic detection.

\sphinxincludegraphics{{ULCMS_24}.png}


\paragraph{Peak Curation}
\label{\detokenize{UnlabeledLCMSWorkflow:peak-curation}}
There are multiple ways to curate peaks in El-MAVEN, though following are the two broad workflows:

\sphinxstylestrong{1. Manual Peak Curation using Compound DB widget}

To use manual curation using the compound DB widget, users have to iterate over all the compounds in the compound DB on the extreme left of the window, as highlighted in the image below.

\sphinxincludegraphics{{ULCMS_25}.png}

Once on a compound, El-MAVEN shows the highest ranked group for that m/z. Users can now choose a group or reject it.

First, users need to double click on the peak group of their choice. This will get the retention time line to the median of the group and also add the metabolite to the bookmarks table (as shown in the image below). Users can read more about the bookmarks table \sphinxhref{https://github.com/ElucidataInc/El-MAVEN/wiki/Introduction-to-El-MAVEN-UI\#5-eic-window}{here}.

\sphinxincludegraphics{{ULCMS_26}.png}

When the users select the first group, they would be asked if they would like to auto-save the state of the application. This feature allows the users to go back to their curated peaks if they so wish in future.

\sphinxincludegraphics{{ULCMS_27}.png}

Qualifying peaks as good or bad is explained in the next few sections.

\sphinxstylestrong{2. Automated Peak Curation}

El-MAVEN can automatically select high intensity and high quality groups. This workflow is called automatic peak curation. To enable this workflow users have to click on the peak detection widget present in the top left of the window. Upon clicking the peak detection widget \sphinxincludegraphics{{Widget_29}.png} the following dialog box will open.

\sphinxincludegraphics{{ULCMS_30}.png}

Users can read more about the peak detection widget \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#peak-detection}{here}.

Upon selecting the default parameters, users can click on \sphinxstyleemphasis{Find Peaks} to select the most important peaks. Once the peak detection is completed, a peak table shows up at the bottom of the window.

\sphinxincludegraphics{{ULCMS_31}.png}

Users can now iterate over these peaks by marking them as good or bad by clicking on the good or bad buttons present in the peak table as shown below.

\sphinxincludegraphics{{ULCMS_32}.png}


\paragraph{Guidelines for Peak Picking}
\label{\detokenize{UnlabeledLCMSWorkflow:guidelines-for-peak-picking}}\begin{itemize}
\item {} 
A peak’s width and shape are two very crucial things to look at while classifying a peak as good or bad. A peak’s shape should have a Gaussian distribution and width should not be spread across a wide range of retention time.

\end{itemize}

\sphinxincludegraphics{{ULCMS_33}.png}
\begin{itemize}
\item {} 
Peak Intensities for a group are plotted as bar plots for all the samples. These bar plots have heights relative to the other samples.Thus, for a good peak the intensities should be high.

\end{itemize}

\sphinxincludegraphics{{ULCMS_34}.png}
\begin{itemize}
\item {} 
Intensity Barplot heights should be higher for all the samples than Blank samples. We use intensities of Blank samples to set our group baseline. Blank intensities are used to calibrate intensity values across zero concentration.

\item {} 
A good peak should have standards with varying intensity in a particular fashion (increasing or decreasing).

\item {} 
Quality Control (QC) samples give us information about the quality of the data, i.e., it assesses reproducibility and software performance. Samples whose intensities and concentrations are already known are used as QCs to determine if the instrument is working as expected. Values (and scales) can be calibrated using QCs.

\item {} 
If peak groups of a particular metabolite are separated apart (not aligned well) then we should use stringent alignment parameters to overcome this problem.

\item {} 
For a particular metabolite, let’s say if it has n number of groups, then the group which is much closer to the above guidelines should be selected as a good peak. Multiple groups can also be selected in case of ambiguity (if retention time information is not provided).

\end{itemize}

A good peak would look similar to the following peaks:

\sphinxincludegraphics{{ULCMS_35}.png}
\begin{itemize}
\item {} 
Gaussian shape

\item {} 
Perfect grouping, narrow retention time

\item {} 
Good sample intensities

\item {} 
Low blank intensities

\item {} 
QCs look good

\item {} 
An observable trend in intensity bars of standards, as well as samples.

\end{itemize}

Some examples of bad peaks are given below:
\begin{itemize}
\item {} 
The peaks do not have a Gaussian shape. Low intensity peaks are not grouping well. QC intensities (10\textasciicircum{}4) are too high with respect to the low sample intensities (10\textasciicircum{}2), which are very close to the noise level.

\end{itemize}

\sphinxincludegraphics{{ULCMS_36}.png}
\begin{itemize}
\item {} 
The peaks have a good Gaussian shape. But the blank intensity bars are high. All the sample intensity bars are shorter or roughly equal to the blank intensities, implying that the peaks are noisy. This should be marked bad if better groups of the same metabolite are available.

\end{itemize}

\sphinxincludegraphics{{ULCMS_37}.png}
\begin{itemize}
\item {} 
The intensity levels are high. The blank intensities are lower. However, the peaks are spread over a long range of retention time, have poor grouping, and have forward trailing peaks. If the signal to noise ratio was improved, this peak would probably not be detected.

\end{itemize}

\sphinxincludegraphics{{ULCMS_38}.png}
\begin{itemize}
\item {} 
In the following image, many sample intensities are missing from the intensities bar plots. Peaks do not have a Gaussian shape, nor good grouping. These peaks are probably noise which have been wrongly annotated. The blank intensities are high as well.

\end{itemize}

\sphinxincludegraphics{{ULCMS_39}.png}
\begin{itemize}
\item {} 
This is a noisy group. There are no discrete peaks visible in the image. The X-axis is crowded with noise. The peak shape is sharp, triangular, or line-like; not Gaussian. The intensity levels are high, but so are the noise levels.

\end{itemize}

\sphinxincludegraphics{{ULCMS_40}.png}
\begin{itemize}
\item {} 
The peaks don’t have a Gaussian shape, and are also noisy. The intensity values are very low.

\end{itemize}

\sphinxincludegraphics{{ULCMS_41}.png}
\begin{itemize}
\item {} 
For low intensity groups like this, the peak characteristics can be determined by zooming in.

\end{itemize}

\sphinxincludegraphics{{ULCMS_42}.png}
\begin{quote}

The mouse can be used to select the area of the peak as shown below
\end{quote}

\sphinxincludegraphics{{ULCMS_43}.png}
\begin{quote}

On zooming, it will be easy to make a decision on peak quality
\end{quote}

\sphinxincludegraphics{{ULCMS_44}.png}

The user can mark any ambiguous peaks as good, and can review all such peaks later in the process.


\paragraph{Export}
\label{\detokenize{UnlabeledLCMSWorkflow:export}}
There are multiple export options available for storing marked peak data. Users can either generate a PDF report to save the EIC for every metabolite, export data for a particular group in .csv format, or export the EICs to a Json file as shown below.

\sphinxincludegraphics{{ULCMS_45}.png}

Users can select \sphinxstyleemphasis{All, Good, Bad or Selected} peaks to export.

\sphinxincludegraphics{{ULCMS_46}.png}

The \sphinxstyleemphasis{Export Groups to CSV} option \sphinxincludegraphics{{Widget_36}.png} lets the users save the ‘good’/’bad’ labels along with the peak table. Users also have the option to filter out rows that have a certain label while exporting the table.

\sphinxstyleemphasis{Generate PDF Report} option \sphinxincludegraphics{{Widget_35}.png} saves all EICs with their corresponding bar plots in a PDF file.

\sphinxstyleemphasis{Export EICs to Json} option \sphinxincludegraphics{{Widget_37}.png} exports all EICs to a Json file.

Another option is to export the peak data in .mzroll format that can be directly loaded into El-MAVEN by clicking on the Load \sphinxstyleemphasis{Samples\textbar{}Projects\textbar{}Peaks} option in the File menu. For this, go to the File option in the menu bar, and click on ‘\sphinxstyleemphasis{Save Project}’.

\sphinxincludegraphics{{ULCMS_47}.png}


\subsubsection{Labeled LC-MS Workflow}
\label{\detokenize{LabeledLCMSWorkflow:labeled-lc-ms-workflow}}\label{\detokenize{LabeledLCMSWorkflow::doc}}
This is a tutorial for processing Labeled LC/MS data files through El-MAVEN.


\paragraph{Preprocessing}
\label{\detokenize{LabeledLCMSWorkflow:preprocessing}}
msConvert is a command-line/GUI tool that is used to convert between various mass spectroscopy data formats, developed and maintained by proteoWizard. Raw data files obtained from mass spectrometers need to be converted to certain acceptable formats before processing in El-MAVEN.

\sphinxstylestrong{Input}

msConvert supports the following formats:
\begin{itemize}
\item {} 
.mzXML

\item {} 
.mzML

\item {} 
.RAW ThermoFisher

\item {} 
.RAW Walters

\item {} 
.d Agilent

\item {} 
.wiff ABSciex

\end{itemize}

The settings used for msConvert as a GUI tool are captured in the following screenshots:

\sphinxincludegraphics{{LLCMS_1}.png}

\begin{sphinxadmonition}{note}{Note:}
zlib compression is enabled by default in msConvert. El-MAVEN in its current form does not support zlib compression. It is important to uncheck “Use zlib compression” box.
\end{sphinxadmonition}

\sphinxstylestrong{Output}

msConvert can convert to an array of different formats but El-MAVEN primarily uses .mzXML and .mzML formats.


\paragraph{Launch El-MAVEN}
\label{\detokenize{LabeledLCMSWorkflow:launch-el-maven}}
Once sample files are ready for processing, launch El-MAVEN.

\sphinxincludegraphics{{LLCMS_2}.png}


\paragraph{Adjust Global Settings}
\label{\detokenize{LabeledLCMSWorkflow:adjust-global-settings}}
Global Settings can be changed from the \sphinxstyleemphasis{Options} dialog \sphinxincludegraphics{{Widget_1}.png}. There are 9 tabs in the dialog. Each of these tabs has parameters related to a different module in El-MAVEN. For example, a tab for Instrumentation details, a tab for File Import settings etc.

\sphinxincludegraphics{{LLCMS_3}.png}

The \sphinxstyleemphasis{m/z} option scans the groups to find any specific m/z value and plot its corresponding EIC. The +/- option to its right is to specify the expected mass resolution error in parts per million (ppm).

\sphinxincludegraphics{{LLCMS_4}.png}

To know more about the functionality of different tabs and their settings, users can see the \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#global-settings}{Widgets page}. Please be sure to set the desired settings before processing an input file.


\paragraph{Load Samples}
\label{\detokenize{LabeledLCMSWorkflow:load-samples}}
Users can go to File in the Menu and click on \sphinxstyleemphasis{Load Samples\textbar{}Projects\textbar{}Peaks} option. Then navigate to the folder containing the sample data, select all .mzXML or .mzML files and click Open. A loading bar displays the progress at the bottom.

\sphinxincludegraphics{{LLCMS_5}.png}

When the samples have loaded, users should see a sample panel on the left side. If it is not displayed automatically, click on the \sphinxstyleemphasis{Show Samples Widget} button on the toolbar. El-MAVEN automatically assigns a color to every sample. Users can select/deselect any sample by clicking the checkbox on the left of the sample name.

\sphinxincludegraphics{{LLCMS_6}.png}


\paragraph{Load Compound Database}
\label{\detokenize{LabeledLCMSWorkflow:load-compound-database}}
Users can click on \sphinxstyleemphasis{Compounds} option in the leftmost menu, navigate to the folder containing the standard database file, select the appropriate .csv file and click \sphinxstyleemphasis{Open}. Alternatively, users may use any of the default files loaded on start-up.

\sphinxincludegraphics{{LLCMS_7}.png}

This is a sample Compound Database:

\sphinxincludegraphics{{LLCMS_8}.png}

It lists all metabolite names, chemical formula, HMDB ID, and the class/category of compounds they belong to (if known).


\paragraph{Mark Blanks}
\label{\detokenize{LabeledLCMSWorkflow:mark-blanks}}
Users can mark the blanks by selecting the blank samples from the list on screen, and clicking on the \sphinxstyleemphasis{Set as a Blank Sample} icon \sphinxincludegraphics{{Widget_10}.png} in Samples menu.

\sphinxincludegraphics{{LLCMS_9}.png}

Multiple blanks can be marked together. The blanks will appear black as shown in the image below:

\sphinxincludegraphics{{LLCMS_10}.png}


\paragraph{Alignment}
\label{\detokenize{LabeledLCMSWorkflow:alignment}}
Prolonged use of the LC column can lead to a drift in retention time across samples. Alignment shifts the peak retention time in every sample to correct for this drift and brings the peaks closer to median retention time of the group.

\sphinxincludegraphics{{LLCMS_11}.png}

In the above image, EIC for a UTP group is displayed. If the samples were aligned, all peaks would lie at the same retention time. Since this is not the case, the samples need to be aligned.

\sphinxstyleemphasis{Alignment visualization} \sphinxincludegraphics{{Widget_26}.png} can be used to judge the extent of deviation from median retention time.

\sphinxincludegraphics{{LLCMS_12}.png}

In the above visualization, each box represents a peak from the selected group at its current retention time. Samples are said to be perfectly aligned when all peak boxes lie on the same vertical axis. The peaks are considerably scattered in the above image and therefore the samples should be aligned for better grouping of peaks.

\sphinxstylestrong{Perform Alignment}

\sphinxstyleemphasis{Alignment settings} can be adjusted using the Align button \sphinxincludegraphics{{Widget_25}.png}. This example was aligned with \sphinxstyleemphasis{Poly fit} algorithm with default parameters.

\sphinxincludegraphics{{LLCMS_13}.png}

Post-alignment the peaks in the group should appear closer to the median retention time of the group.

\sphinxincludegraphics{{LLCMS_14}.png}

\sphinxincludegraphics{{LLCMS_15}.png}

Pre-alignment, the peaks were considerably scattered while the aligned peaks lie nearly on the same axis. Users can run alignment again with different parameters if required (or with a different algorithm). Further details on Alignment settings are available on the \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#alignment}{Widgets page}.


\paragraph{Peak Grouping}
\label{\detokenize{LabeledLCMSWorkflow:peak-grouping}}
Peak grouping is an integral part of the El-MAVEN workflow that categorizes all detected peaks into groups on the basis of certain user-controlled parameters. A group score is calculated for every peak during the process. The formula for this score takes into account the difference in retention time, intensities between peaks (smaller difference leads to a better score) and any existing overlap between them (higher extent of overlap leads to better score). All three parameters have certain weights attached to them that can be controlled by the users. The formula for the score is shown in the image. More details on it can be found on the \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#global-settings}{Widgets page}.

\sphinxincludegraphics{{LLCMS_16}.png}

\sphinxincludegraphics{{LLCMS_17}.png}

The above image shows two groups in the EIC window. The highlighted (solid circles) peaks belong to group A, the peaks to its left with empty circles belong to another group B. The short peaks in group A that are close to the baseline and peaks in group B come from the same samples. Additionally, the high intensity peaks of group A have a similar peak shape to group B peaks. These peaks might have been wrongly classified into separate groups because of the difference in retention time range of the two sets of peaks. The weights attached to difference in retention time and intensities, and extent of overlap can be adjusted for better grouping.

Grouping parameters can be changed from the Options dialog \sphinxincludegraphics{{Widget_1}.png}.

\sphinxincludegraphics{{LLCMS_18}.png}

\sphinxincludegraphics{{LLCMS_19}.png}

Giving less priority to difference in retention time and intensities results in the two groups being merged into a single  group while the peaks that lay close to the baseline are no longer classified as valid peaks.


\paragraph{Baseline}
\label{\detokenize{LabeledLCMSWorkflow:baseline}}
When measuring a number of peaks, it is often more effective to subtract an estimated baseline from the data. This baseline should be set where ideally no peaks occur. Although sometimes the program sets a particular baseline such that one or more peaks occur below that baseline value. In the following image, the dashed line represents each baseline:

\sphinxincludegraphics{{LLCMS_20}.png}

The corresponding peaks are indicated with solid circles:

\sphinxincludegraphics{{LLCMS_21}.png}

The baseline correction can be done in the \sphinxstyleemphasis{Peak Detection} tab by clicking on \sphinxstyleemphasis{Options} button:

\sphinxincludegraphics{{LLCMS_22}.png}

Further details on settings can be accessed \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#peak-detection}{here}.


\paragraph{Isotope Detection}
\label{\detokenize{LabeledLCMSWorkflow:isotope-detection}}
\sphinxstylestrong{Samples are labeled?}

The \sphinxstyleemphasis{Peaks} dialog \sphinxincludegraphics{{Widget_29}.png} can be used to detect labeled peaks along with the unlabeled ones in the Peaks Table.

\sphinxincludegraphics{{LLCMS_23}.png}

On opening the \sphinxstyleemphasis{Feature Detection Selection} tab, the \sphinxstyleemphasis{Report Isotopic Peaks} box must be checked. Clicking on the \sphinxstyleemphasis{Isotope Detection Options} gives the following window. Alternately, these settings can also be accessed from the \sphinxstyleemphasis{Options} dialog.

\sphinxincludegraphics{{LLCMS_24}.png}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Bookmarks, peak detection, file export}: To select the labeled atoms that should be used in bookmarking, peak detection and export. D2: Deuterium, C13: Labeled carbon, N15: Labeled nitrogen, S34: Labeled sulphur.

\item {} 
\sphinxstyleemphasis{Isotopic widget}: To select the labeled atoms that should be displayed in the isotopic widget. D2: Deuterium, C13: Labeled carbon, N15: Labeled nitrogen, S34: Labeled sulphur.

\item {} 
\sphinxstyleemphasis{Number of M+n isotopes}: To set the maximum number of labeled atoms per ion in the experiment.

\item {} 
\sphinxstyleemphasis{Abundance Threshold}: To set the minimum threshold for isotopic abundance. Isotopic abundance is the ratio of intensity of isotopic peak over the parent peak.

\end{itemize}

\sphinxstylestrong{Filter Isotopic Peaks based on these criteria}
\begin{itemize}
\item {} 
\sphinxstyleemphasis{Minimum Isotope-Parent Correlation}: To set the minimum threshold for isotope-parent peak correlation. This correlation is a measure of how often they appear together.

\item {} 
\sphinxstyleemphasis{Isotope is within {[}X{]} scans of parent}: To set the maximum scan difference between isotopic and parent peaks. This is a measure of how closely they appear together on the retention time scale.

\item {} 
\sphinxstyleemphasis{Maximum \% Error to Natural Abundance}: To set the maximum natural abundance error expected. Natural abundance of an isotope is the expected ratio of amount of isotope over the amount of parent molecule in nature. Error is the difference between observed and natural abundance as a fraction of natural abundance.

\item {} 
\sphinxstyleemphasis{Correct for Natural C13 Isotope Abundance}: The box should be checked to correct for natural C13 abundance.

\end{itemize}

In the image below, Peak Table 3 has a drop down button with metabolites that shows all labeled isotopologues of that particular metabolite.

\sphinxincludegraphics{{LLCMS_25}.png}

\sphinxincludegraphics{{LLCMS_26}.png}

\sphinxincludegraphics{{LLCMS_27}.png}

\sphinxstylestrong{Show Isotope Plots}

This icon \sphinxincludegraphics{{Widget_23}.png} on top displays the isotope plots for a group. The red colored bar plot for UTP group is shown below. Each bar in the plot represents the relative percentage of different isotopic species for the selected group in a sample.

\sphinxincludegraphics{{LLCMS_28}.png}


\paragraph{Mass Spectra}
\label{\detokenize{LabeledLCMSWorkflow:mass-spectra}}
Mass Spectra Widget \sphinxincludegraphics{{Widget_38}.png} displays each peak, its mass, and intensity for a scan. As the widget shows all detected masses in a scan, the ppm window for the EIC and consequently grouping can be adjusted accordingly. This feature is especially useful for MS/MS data and isotopic detection.

\sphinxincludegraphics{{LLCMS_29}.png}


\paragraph{Peak Curation}
\label{\detokenize{LabeledLCMSWorkflow:peak-curation}}
Generally there are two broad workflows to curate peaks in El-MAVEN:
\begin{itemize}
\item {} 
Manual Peak Curation using Compound DB widget

\item {} 
Automated Peak Curation

\end{itemize}

Although, for labeled data Automatic Peak Curation is not meaningful because it will not curate any labeled groups.

\sphinxstylestrong{Manual Peak Curation using Compound DB widget}

Clicking the \sphinxstyleemphasis{Peaks} icon \sphinxincludegraphics{{Widget_29}.png} on the top opens the settings dialog.

\sphinxincludegraphics{{LLCMS_30}.png}

Users must check the box for \sphinxstyleemphasis{Report Isotopic Peaks} in the \sphinxstyleemphasis{Group Filtering} tab.

\begin{sphinxadmonition}{note}{Note:}
Users should not click on \sphinxstyleemphasis{Find Peaks} after checking the box for manual curation. Clicking on that option would start Automatic Peak Detection. For adjusting other settings, users can access them through the Options \sphinxincludegraphics{{Widget_1}.png} dialog .
\end{sphinxadmonition}

For more details on how to access Peak Detection settings, read this \sphinxhref{https://github.com/ElucidataInc/ElMaven/wiki/Introduction-to-ElMaven-UI\#peak-detection}{Widgets page}.

To use manual curation using the compound DB widget, users have to iterate over all the compounds in the compound DB on the extreme left of the window, as highlighted in the images below.

\sphinxincludegraphics{{LLCMS_31}.png}

\sphinxincludegraphics{{LLCMS_32}.png}

Once on a compound, El-MAVEN shows the highest ranked group for that m/z. Users can now choose a group or reject it. There are two ways to do this.

In the first workflow, users need to double click on the peak group of his choice. This will get the retention time line to the median of the group and also add the metabolite to the bookmarks table (as shown in the image below). Users can read more about the bookmarks table \sphinxhref{https://github.com/ElucidataInc/ElMaven/wiki/Introduction-to-ElMaven-UI\#5-eic-window}{here}.

\sphinxincludegraphics{{LLCMS_33}.png}

\sphinxincludegraphics{{LLCMS_34}.png}

When the users select the first group they would be asked if they would like to auto-save the state of the application. This feature allows the users to go back to their curated peaks if they so wishes in future.

\sphinxincludegraphics{{LLCMS_35}.png}

Users can then use dropdown arrow on bookmarked group to mark all its isotopologues as good or bad.

\sphinxincludegraphics{{LLCMS_36}.png}

After marking all the groups for a compound, users can scroll down to the next compound and decide on the basis of shown EIC if the group should be marked for curation.

\sphinxincludegraphics{{LLCMS_37}.png}

Double clicking on any peak (solid coloured circle) moves the retention time line along the group. And the group moves to the bookmark table.

\sphinxincludegraphics{{LLCMS_38}.png}

Qualifying peaks as good or bad is explained in the next section.


\paragraph{Guidelines for Peak Picking}
\label{\detokenize{LabeledLCMSWorkflow:guidelines-for-peak-picking}}\begin{itemize}
\item {} 
A peak’s width and shape are two very crucial things to look at while classifying a peak as good or bad. The peak’s shape should have a Gaussian distribution and width should not be spread across a wide range of retention time.

\end{itemize}

\sphinxincludegraphics{{LLCMS_39}.png}
\begin{itemize}
\item {} 
Peak Intensities for a group are plotted as bar plots for all the samples. These bar plots have heights relative to the other samples.Thus, for a good peak the intensities should be high.

\end{itemize}

\sphinxincludegraphics{{LLCMS_40}.png}
\begin{itemize}
\item {} 
Intensity Barplot heights should be higher for all the samples than Blank samples, as shown above. We use intensities of Blank samples to set our group baseline. Blank intensities are used to calibrate intensity values across zero concentration.

\item {} 
A good peak should have standards with varying intensity in a particular fashion (increasing or decreasing).

\item {} 
Quality Control (QC) samples give us information about the quality of the data, i.e., it assesses reproducibility and software performance. Samples whose intensities and concentrations are already known are used as QCs to determine if the instrument are working as expected. Values (and scales) can be calibrated using QCs.

\item {} 
If peak groups of a particular metabolite are separated apart (not aligned well) then we should use stringent alignment parameters to overcome this problem.

\item {} 
For a particular metabolite, let’s say if it has n number of groups, then the group which is much closer to the above guidelines should be selected as good peak. Multiple groups can also be selected in case of ambiguity (if retention time information is not provided).

\end{itemize}

\sphinxstylestrong{A good peak would look similar to the following peaks:}

\sphinxincludegraphics{{LLCMS_41}.png}

\sphinxincludegraphics{{LLCMS_42}.png}

\sphinxincludegraphics{{LLCMS_43}.png}
\begin{itemize}
\item {} 
Gaussian shape

\item {} 
Perfect grouping, narrow retention time

\item {} 
Good sample intensities

\item {} 
Low blank intensities

\item {} 
QCs look good

\item {} 
An observable trend in intensity bars of standards, as well as samples.

\end{itemize}

\sphinxstylestrong{Some examples of bad peaks are given below:}
\begin{itemize}
\item {} 
The peaks have a good Gaussian shape. But the blank intensity bars are high. All the sample intensity bars are shorter or roughly equal to the blank intensities, implying the peaks are most likely noise.

\end{itemize}

\sphinxincludegraphics{{LLCMS_44}.png}
\begin{itemize}
\item {} 
The intensity levels are low relatively. The peaks are spread over a long range of retention time. They have poor shape, poor grouping and lie close to noise. If the signal to noise ratio was improved, this peak would probably not be detected.

\end{itemize}

\sphinxincludegraphics{{LLCMS_45}.png}
\begin{itemize}
\item {} 
In the following image, many sample intensities are missing from the intensities bar plots. Peaks do not have a Gaussian shape, nor good grouping. These peaks are probably noise which have been wrongly annotated.

\end{itemize}

\sphinxincludegraphics{{LLCMS_46}.png}
\begin{itemize}
\item {} 
This is a noisy group. There are no discrete peaks visible in the image. The X-axis is crowded with noise. The peak shape is sharp, triangular, or line-like; not Gaussian. The intensity levels are high, but so are noise levels.

\end{itemize}

\sphinxincludegraphics{{LLCMS_47}.png}
\begin{quote}

More examples of noisy peaks:
\end{quote}

\sphinxincludegraphics{{LLCMS_48}.png}

\sphinxincludegraphics{{LLCMS_49}.png}
\begin{itemize}
\item {} 
The peaks don’t have a Gaussian shape, and are also noisy. The intensity values are very low.

\item {} 
For low intensity groups like this, the peak characteristics can be determined by zooming in.

\end{itemize}

\sphinxincludegraphics{{LLCMS_50}.png}
\begin{quote}

The mouse can be used to select the area of the peak as shown below
\end{quote}

\sphinxincludegraphics{{LLCMS_51}.png}
\begin{quote}

On zooming, it will be easy to make a decision on peak quality
\end{quote}

\sphinxincludegraphics{{LLCMS_52}.png}

\begin{sphinxadmonition}{note}{Note:}
Users can mark any ambiguous peaks as good, and can review all such peaks later in the process.
\end{sphinxadmonition}


\paragraph{Export}
\label{\detokenize{LabeledLCMSWorkflow:export}}
There are multiple export options available for storing marked peak data. Users can either generate a PDF report to save the EIC for every metabolite, export data for a particular group in .csv format, or export the EICs to a Json file as shown below.

\sphinxincludegraphics{{LLCMS_53}.png}

Users can select \sphinxstyleemphasis{All, Good, Bad or Selected} peaks to export.

\sphinxincludegraphics{{LLCMS_54}.png}

The \sphinxstyleemphasis{Export Groups to CSV} option \sphinxincludegraphics{{Widget_36}.png} lets the users save the ‘good’/’bad’ labels along with the peak table. Users also have the option to filter out rows that have a certain label while exporting the table.

\sphinxstyleemphasis{Generate PDF Report} option \sphinxincludegraphics{{Widget_35}.png} saves all EICs with their corresponding bar plots in a PDF file.

\sphinxstyleemphasis{Export EICs to Json} option \sphinxincludegraphics{{Widget_37}.png} exports all EICs to a Json file.

Another option is to export the peak data in .mzroll format that can be directly loaded into El-MAVEN by clicking on the Load \sphinxstyleemphasis{Samples\textbar{}Projects\textbar{}Peaks} option in the File menu. For this, go to the File option in the menu bar, and click on ‘\sphinxstyleemphasis{Save Project}’.

\sphinxincludegraphics{{LLCMS_55}.png}


\subsubsection{Labeled LC-MS/MS Workflow}
\label{\detokenize{LabeledLCMSMSWorkflow:labeled-lc-ms-ms-workflow}}\label{\detokenize{LabeledLCMSMSWorkflow::doc}}
This is a tutorial for processing LC/MSMS data files through El-MAVEN.


\paragraph{Preprocessing}
\label{\detokenize{LabeledLCMSMSWorkflow:preprocessing}}
msConvert is a command-line/GUI tool that is used to convert between various mass spectroscopy data formats, developed and maintained by proteoWizard. Raw data files obtained from mass spectrometers need to be converted to certain acceptable formats before processing in El-MAVEN.

\sphinxstylestrong{Input}

msConvert supports the following formats:
\begin{itemize}
\item {} 
.mzXML

\item {} 
.mzML

\item {} 
.RAW ThermoFisher

\item {} 
.RAW Walters

\item {} 
.d Agilent

\item {} 
.wiff ABSciex

\end{itemize}

The settings used for msConvert as a GUI tool are captured in the following screenshot:

\sphinxincludegraphics{{LCMSMS_1}.png}

\begin{sphinxadmonition}{note}{Note:}
It is important that zlib compression is enabled by default in msConvert. El-MAVEN in its current form does not support zlib compression. Make sure to uncheck “Use zlib compression” box.
\end{sphinxadmonition}

\sphinxstylestrong{Output}

msConvert can convert to an array of different formats but El-MAVEN primarily uses .mzXML and .mzML formats.


\paragraph{Launch El-MAVEN}
\label{\detokenize{LabeledLCMSMSWorkflow:launch-el-maven}}
Once sample files are ready for processing, launch El-MAVEN.

\sphinxincludegraphics{{LCMSMS_2}.png}


\paragraph{Adjust Global Settings}
\label{\detokenize{LabeledLCMSMSWorkflow:adjust-global-settings}}
Global Settings can be changed from the \sphinxstyleemphasis{Options} dialog \sphinxincludegraphics{{Widget_1}.png}. There are 9 tabs in the dialog. Each of these tabs has parameters related to a different module in El-MAVEN. For example, a tab for Instrumentation details, a tab for file import settings, etc.

\sphinxincludegraphics{{LCMSMS_3}.png}

To know more about the functionality of different tabs and their settings, users can see the \sphinxhref{https://elmaven.readthedocs.io/en/develop/IntroductiontoElMAVENUI.html\#global-settings}{Widgets page}. Please be sure to set the desired settings before processing an input file.


\paragraph{Load Samples}
\label{\detokenize{LabeledLCMSMSWorkflow:load-samples}}
Users can go to File in the Menu, click on \sphinxstyleemphasis{Load Samples\textbar{}Projects\textbar{}Peaks} option in the File menu. Then navigate to the folder containing the sample data, select all .mzXML or .mzml files and click Open. A loading bar displays the progress at the bottom.

\sphinxincludegraphics{{LCMSMS_4}.png}

When the samples have loaded, users should see a sample panel on the left side. If it is not displayed automatically, click on the \sphinxstyleemphasis{Show Samples Widget} button on the toolbar. El-MAVEN automatically assigns a color to every sample. Users can select/deselect any sample by clicking the checkbox on the left of the sample name.

\sphinxincludegraphics{{LCMSMS_5}.png}


\paragraph{Load Compound Database}
\label{\detokenize{LabeledLCMSMSWorkflow:load-compound-database}}
Users can click on \sphinxstyleemphasis{Compounds} option in the leftmost menu, navigate to the folder containing the standard database file, select the appropriate .csv file and click \sphinxstyleemphasis{Open}. Alternatively, users may use any of the default files loaded on start-up.

\sphinxincludegraphics{{LCMSMS_6}.png}

This is a sample Compound Database:

\sphinxincludegraphics{{LCMSMS_7}.png}

It lists all metabolite names, chemical formula, HMDB ID, and the class/category of compounds they belong to (if known).


\paragraph{Mark Blanks}
\label{\detokenize{LabeledLCMSMSWorkflow:mark-blanks}}
Users can mark the blanks by selecting the blank samples from the list on screen, and clicking on the \sphinxstyleemphasis{Set as a Blank Sample} icon \sphinxincludegraphics{{Widget_10}.png} in Samples menu.

\sphinxincludegraphics{{LCMSMS_8}.png}

Multiple blanks can be marked together. The blanks will appear black as shown in the image below:

\sphinxincludegraphics{{LCMSMS_9}.png}


\paragraph{Alignment}
\label{\detokenize{LabeledLCMSMSWorkflow:alignment}}
(missing)


\paragraph{Export}
\label{\detokenize{LabeledLCMSMSWorkflow:export}}
There are multiple export options available for storing marked peak data. Users can either generate a PDF report to save the EIC for every metabolite, export data for a particular group in .csv format, or export the EICs to a Json file as shown below.

\sphinxincludegraphics{{LCMSMS_10}.png}

Users can select \sphinxstyleemphasis{All, Good, Bad or Selected} peaks to export.

\sphinxincludegraphics{{LCMSMS_11}.png}

The \sphinxstyleemphasis{Export Groups to CSV} option \sphinxincludegraphics{{Widget_36}.png} lets the users save the ‘good’/’bad’ labels along with the peak table. Users also have the option to filter out rows that have a certain label while exporting the table.

\sphinxstyleemphasis{Generate PDF Report} option \sphinxincludegraphics{{Widget_35}.png} saves all EICs with their corresponding bar plots in a PDF file.

\sphinxstyleemphasis{Export EICs to Json} option \sphinxincludegraphics{{Widget_37}.png} exports all EICs to a Json file.

Another option is to export the peak data in .mzroll format that can be directly loaded into El-MAVEN by clicking on the Load \sphinxstyleemphasis{Samples\textbar{}Projects\textbar{}Peaks} option in the File menu. For this, go to the File option in the menu bar, and click on ‘\sphinxstyleemphasis{Save Project}’.

\sphinxincludegraphics{{LCMSMS_12}.png}


\subsection{Validation}
\label{\detokenize{Documentation:validation}}

\subsubsection{Labeled MS/MS Validation}
\label{\detokenize{LabeledMSMSValidation:labeled-ms-ms-validation}}\label{\detokenize{LabeledMSMSValidation::doc}}

\paragraph{MS/MS Support in El-MAVEN}
\label{\detokenize{LabeledMSMSValidation:ms-ms-support-in-el-maven}}
Tandem mass spectrometry or MS/MS is an important technique in analytical Mass spectrometry. Processing MS/MS data for a large batch of samples can be a time-consuming task. We have added some support for processing such data in El-MAVEN.

Data acquired using the following methods are currently supported in El-MAVEN: Multiple Reaction Monitoring, Parallel Reaction Monitoring (or Full-Scan MS2) and Data Dependant MS2.


\paragraph{Feature Additions}
\label{\detokenize{LabeledMSMSValidation:feature-additions}}\begin{itemize}
\item {} 
Text search option using precursor and product m/z

\item {} 
Targeted Peak Picking for MS/MS

\item {} 
Manual annotation of SRM transition

\end{itemize}


\paragraph{Validation}
\label{\detokenize{LabeledMSMSValidation:validation}}
In order to ensure correct reading and processing of MS2 sample files in El-MAVEN, MS spectra and chromatograms from El-MAVEN were validated using \sphinxhref{https://skyline.ms/wiki/home/software/Skyline/page.view?name=default}{Skyline} by ProteoWizard.


\paragraph{Mass Spectra Validation}
\label{\detokenize{LabeledMSMSValidation:mass-spectra-validation}}
Dataset: Full-scan MS2 sample file


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|}
\hline
\sphinxstyletheadfamily 
Fragment
&\sphinxstyletheadfamily 
Skyline
&\sphinxstyletheadfamily 
El-MAVEN
\\
\hline
\#1
&
\sphinxincludegraphics{{Validation_1}.png}
&
\sphinxincludegraphics{{Validation_2}.png}
\\
\hline
\#2
&
\sphinxincludegraphics{{Validation_3}.png}
&
\sphinxincludegraphics{{Validation_4}.png}
\\
\hline
\#3
&
\sphinxincludegraphics{{Validation_5}.png}
&
\sphinxincludegraphics{{Validation_6}.png}
\\
\hline
\#4
&
\sphinxincludegraphics{{Validation_7}.png}
&
\sphinxincludegraphics{{Validation_8}.png}
\\
\hline
\#5
&
\sphinxincludegraphics{{Validation_9}.png}
&
\sphinxincludegraphics{{Validation_10}.png}
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}


\paragraph{Chromatogram Validation}
\label{\detokenize{LabeledMSMSValidation:chromatogram-validation}}
Dataset: Multiple Reaction Monitoring


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|}
\hline
\sphinxstyletheadfamily 
Fragment
&\sphinxstyletheadfamily 
Skyline
&\sphinxstyletheadfamily 
El-MAVEN
\\
\hline
\#1
&
\sphinxincludegraphics{{Validation_11}.png}
&
\sphinxincludegraphics{{Validation_12}.png}
\\
\hline
\#2
&
\sphinxincludegraphics{{Validation_13}.png}
&
\sphinxincludegraphics{{Validation_14}.png}
\\
\hline
\#3
&
\sphinxincludegraphics{{Validation_15}.png}
&
\sphinxincludegraphics{{Validation_16}.png}
\\
\hline
\#4
&
\sphinxincludegraphics{{Validation_17}.png}
&
\sphinxincludegraphics{{Validation_18}.png}
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}

Multiple Transition cases: The same precursor/product pairs can be tracked in multiple runs, for example, in cases where two metabolites have common fragments like Pyruvate 89/89 and Lactate 89/89. El-MAVEN lists these runs separately in the SRM List widget. Following are some examples:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|T|}
\hline
\sphinxstyletheadfamily 
Fragment
&\sphinxstyletheadfamily 
Skyline
&\sphinxstyletheadfamily 
El-MAVEN
&\sphinxstyletheadfamily 
Comments
\\
\hline
Lactate
Fragment \#1
&
\sphinxincludegraphics{{Validation_19}.png}
&
Compound widget:

\sphinxincludegraphics{{Validation_20}.png}

Both runs for this
fragment have been
annotated as
Lactate in the SRM
widget

Transition 1:

\sphinxincludegraphics{{Validation_21}.png}

Transition 2:

\sphinxincludegraphics{{Validation_22}.png}
&
Skyline EIC matches
Transition 2 from
the SRM widget in
El-MAVEN
\\
\hline
Pyruvate
Fragment \#1
&
EIC is same as
that of Lactate
for this fragment.
Different peak is
selected according
to the expected
retention time

\sphinxincludegraphics{{Validation_23}.png}
&
Compound widget:

\sphinxincludegraphics{{Validation_24}.png}

Zoomed in:

\sphinxincludegraphics{{Validation_25}.png}

SRM widget:
No transitions
mapped to
Pyruvate
&
Unable to find
record of another
transition in
Skyline. The choppy
peaks in El-MAVEN
are due to the
merging of two
transitions
annotated as
Lactate
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}

Text search/Compound widget mismatch: Using the text search feature for MRM data can result in choppy peaks at times as compared to those from the SRM/compound widget. Following are some examples:


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|T|}
\hline
\sphinxstyletheadfamily 
Fragment
&\sphinxstyletheadfamily 
Skyline
&\sphinxstyletheadfamily 
El-MAVEN Text
Search
&\sphinxstyletheadfamily 
El-MAVEN Compound
Widget
\\
\hline
\#1
&
\sphinxincludegraphics{{Validation_26}.png}
&
\sphinxincludegraphics{{Validation_27}.png}
&
\sphinxincludegraphics{{Validation_28}.png}
\\
\hline
\#2
&
\sphinxincludegraphics{{Validation_29}.png}
&
\sphinxincludegraphics{{Validation_30}.png}
&
\sphinxincludegraphics{{Validation_31}.png}
\\
\hline
\#3
&
\sphinxincludegraphics{{Validation_32}.png}
&
\sphinxincludegraphics{{Validation_33}.png}
&
\sphinxincludegraphics{{Validation_34}.png}
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}

Dataset: Full-scan MS2


\begin{savenotes}\sphinxattablestart
\centering
\begin{tabulary}{\linewidth}[t]{|T|T|T|T|T|}
\hline
\sphinxstyletheadfamily 
Fragment
&\sphinxstyletheadfamily 
Skyline
&\sphinxstyletheadfamily 
El-MAVEN Text
Search
&\sphinxstyletheadfamily 
El-MAVEN Compund
Widget
&\sphinxstyletheadfamily 
Comments
\\
\hline
\#1
&
\sphinxincludegraphics{{Validation_32}.png}
&
\sphinxincludegraphics{{Validation_33}.png}
&
\sphinxincludegraphics{{Validation_34}.png}
&
All three EIC’s
match exactly
\\
\hline
\#2
&
\sphinxincludegraphics{{Validation_35}.png}
&
\sphinxincludegraphics{{Validation_36}.png}
&
\sphinxincludegraphics{{Validation_37}.png}
&
All three EIC’s
match exactly
\\
\hline
\#3
&
\sphinxincludegraphics{{Validation_38}.png}
&
\sphinxincludegraphics{{Validation_39}.png}
&
\sphinxincludegraphics{{Validation_40}.png}
&
EIC from the
compound widget is
different as one of
the runs has been
annotated as this
fragment which
pulls up the EIC
for the whole run
\\
\hline
\end{tabulary}
\par
\sphinxattableend\end{savenotes}


\paragraph{Issues Observed}
\label{\detokenize{LabeledMSMSValidation:issues-observed}}\begin{itemize}
\item {} 
Text search and compound/SRM widget EIC mismatch for MRM data: EIC obtained from the compound widget and Skyline are the same. The data points on the plot are same for all three. The error in text search is due to different processing. This has been filed as issue number \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/487}{\#487}; Status: Unresolved

\item {} 
Text search and compound widget EIC mismatch for PRM data: EIC from Text search and Skyline are the same. The error in compound widget happens due to linking of SRM ID to the compound widget. This pulls up the whole run instead of the particular fragment. This is a known issue \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/442}{\#442}; Status: Unresolved

\item {} 
Merged peaks in case of multiple transitions: This is being handled as part of issue number \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/405}{\#405}; Status: In progress

\item {} 
Mass spectra displays wrong product M/z value in some cases: The prodMz field in the title of spectra widget displays the base peak m/z value which may or may not be the same as the fragment m/z. This will be resolved as part of \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/396}{\#396}; Status: Unresolved

\item {} 
Peak table displays NA in Expected m/z column for MRM data: This has been filed as issue number \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/493}{\#493}

\end{itemize}


\paragraph{Conclusion}
\label{\detokenize{LabeledMSMSValidation:conclusion}}
Mass spectra validation against Skyline passed. This indicates correct parsing of MS/MS data in El-MAVEN. EIC validation against Skyline highlighted some known and unknown issues. We recommend using Text search for PRM data and Compound/SRM widget for MRM data processing until the reported issues have been fixed.


\section{Contributors}
\label{\detokenize{Contributors:contributors}}\label{\detokenize{Contributors::doc}}\begin{itemize}
\item {} 
\sphinxhref{http://genomics-pubs.princeton.edu/mzroll/index.php}{MAVEN team at Princeton University}

\item {} 
\sphinxhref{https://www.calicolabs.com/team-member/eugene-melamud/}{Eugene Melamud}

\item {} 
\sphinxhref{https://github.com/chubukov}{Victor Chubukov}

\item {} 
\sphinxhref{https://github.com/GeorgeSabu}{George Sabu}

\item {} 
\sphinxhref{https://github.com/sahil21}{Sahil}

\item {} 
\sphinxhref{https://github.com/Raghavdata}{Raghav Sehgal}

\item {} 
\sphinxhref{https://github.com/shubhra-agrawal}{Shubhra Agrawal}

\item {} 
\sphinxhref{https://github.com/r-el-maya}{Raghuram Reddy}

\item {} 
\sphinxhref{https://github.com/rish9511}{Rishabh Gupta}

\item {} 
\sphinxhref{https://github.com/IpankajI}{Pankaj Kumar}

\item {} 
\sphinxhref{https://github.com/rkdahmiwal}{Rahul Kumar}

\item {} 
\sphinxhref{https://github.com/Giridhari013}{Giridhari}

\item {} 
\sphinxhref{https://github.com/kiranvarghese2}{Kiran Varghese}

\item {} 
\sphinxhref{https://github.com/naman}{Naman Gupta}

\item {} 
\sphinxhref{https://github.com/francisglee}{Francis Lee}

\item {} 
\sphinxhref{https://github.com/avijitzutshi}{Avijit Zutshi}

\end{itemize}


\section{Contributing}
\label{\detokenize{Contributing:contributing}}\label{\detokenize{Contributing::doc}}
You are welcome to contribute. Please go through our \sphinxhref{https://github.com/ElucidataInc/ElMaven/blob/develop/CONTRIBUTING.rst}{contributing guidelines} and \sphinxhref{https://github.com/ElucidataInc/ElMaven/blob/develop/CODE\_OF\_CONDUCT.rst}{code of conduct}. These guidelines include directions for coding standards, filing issues and development guidelines.


\section{References}
\label{\detokenize{References:references}}\label{\detokenize{References::doc}}
To understand \sphinxhref{http://genomics-pubs.princeton.edu/mzroll/index.php}{MAVEN’s} and \sphinxhref{https://elucidatainc.github.io/ElMaven/}{El-MAVEN’s} workflows and features, please refer to the following literature on Maven:
\begin{itemize}
\item {} 
\sphinxhref{https://pubs.acs.org/doi/abs/10.1021/ac1021166}{Metabolomic Analysis and Visualization Engine for LC-MS Data}, Eugene Melamud, Livia Vastag, and Joshua D. Rabinowitz, Analytical Chemistry 2010 82 (23), 9818-9826

\item {} 
\sphinxhref{https://currentprotocols.onlinelibrary.wiley.com/doi/abs/10.1002/0471250953.bi1411s37}{LC-MS Data Processing with MAVEN: A Metabolomic Analysis and Visualization Engine}, Clasquin, M. F., Melamud, E. and Rabinowitz, J. D. 2012, Current Protocols in Bioinformatics. 37:14.11.1-14.11.23.

\end{itemize}


\section{Acknowledgement}
\label{\detokenize{Acknowledgement:acknowledgement}}\label{\detokenize{Acknowledgement::doc}}
El-MAVEN would not have been possible without the unwavering support, constant feedback and financial support from \sphinxhref{http://www.agios.com/}{Agios Pharmaceuticals, Inc.} El-MAVEN thanks the metabolomics community for its immense contribution in taking the tool forward and making it a great success.


\section{Release History}
\label{\detokenize{ReleaseHistory:release-history}}\label{\detokenize{ReleaseHistory::doc}}

\subsection{Current Release}
\label{\detokenize{ReleaseHistory:current-release}}
\sphinxhref{https://zenodo.org/record/1332034}{Version 0.4.1}

Publication Date: August 4$^{\text{th}}$, 2018
\begin{itemize}
\item {} 
Optimisations
\begin{itemize}
\item {} 
Faster sample upload

\item {} 
Checking/unchecking a sample in the presence of a large peak table is faster \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/723}{(\#723)}

\item {} 
No lag in switching between two large peak tables

\end{itemize}

\item {} 
UI Changes
\begin{itemize}
\item {} 
Added a close button for Scatter Plot widget

\item {} 
Removed ‘Load Peaks’ from Peak tables

\item {} 
‘Gallery’ widget is only available on the side bar panel

\item {} 
Removed ‘Ratio’ and ‘p-value’ columns from Peak tables (available in Scatter Plot Peak Table)

\item {} 
Removed ‘Good’ and ‘Bad’ marking options from EIC widget

\item {} 
Polly upload button is not clickable during the upload process \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/805}{(\#805)}

\item {} 
Single button for exporting EIC on EIC widget

\item {} 
Removed ‘Scatter Plot’ button from side bar panel

\end{itemize}

\item {} 
Bugs
\begin{itemize}
\item {} 
Isotopic peak filtering based on peak quality has been fixed \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/772}{(\#772)}

\item {} 
Crash in bookmark button has been fixed \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/768}{(\#768)}

\item {} 
Empty tables will no longer be uploaded to Polly \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/777}{(\#777)}

\item {} 
Only a single compound database file (.csv) will be uploaded to Polly

\end{itemize}

\end{itemize}


\subsection{Past Releases}
\label{\detokenize{ReleaseHistory:past-releases}}
\sphinxhref{https://zenodo.org/record/1312704}{Version 0.4.0-beta.1}

Publication Date: July 16$^{\text{th}}$, 2018
\begin{itemize}
\item {} 
Fixed Isotopic detection in shift-drag integration \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/781}{(\#781)}

\item {} 
Groups with zero peaks will not be bookmarked on shift-drag integration

\item {} 
Fixed a bug in changing compound database from the drop-down

\item {} 
Getting Started window will not open behind the main application \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/775}{(\#775)}

\end{itemize}

\sphinxhref{https://zenodo.org/record/1305465}{Version 0.4.0-beta}

Publication Date: July 5$^{\text{th}}$, 2018

\sphinxhref{https://zenodo.org/record/1248658}{Version 0.3.2}

Publication Date: May 17$^{\text{th}}$, 2018

\sphinxhref{https://zenodo.org/record/1232373}{Version 0.3.1}

Publication Date: April 27$^{\text{th}}$, 2018
\begin{itemize}
\item {} 
Bug Fixes:
\begin{itemize}
\item {} 
Fixed compound name in CSV export \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues/635}{(\#635)}

\item {} 
Fixed automated detection of high ranking groups

\item {} 
Fix installer issue on Mac (Installer not working due to netcdf addition)

\end{itemize}

\item {} 
UI:
\begin{itemize}
\item {} 
Change default tab for El-MAVEN - Polly Integration dialog

\end{itemize}

\end{itemize}

\sphinxhref{https://zenodo.org/record/1230370}{Version 0.3.1}

Publication Date: April 26$^{\text{th}}$, 2018

\sphinxhref{https://zenodo.org/record/1228065}{Version 0.3.0}

Publication Date: April 24$^{\text{th}}$, 2018

\sphinxhref{https://zenodo.org/record/1227187}{Version 0.3.0}

Publication Date: April 23$^{\text{rd}}$, 2018

\sphinxhref{https://zenodo.org/record/1216928}{Version 0.3.0-beta}

Publication Date: April 11$^{\text{th}}$, 2018
\begin{itemize}
\item {} 
Features
\begin{itemize}
\item {} 
Export Scatter plot groups into a peak table

\item {} 
Introduced Exception handling in sample upload process

\item {} 
Send logs to sentry in the event of a crash

\item {} 
Upload data to Polly through CLI and GUI

\item {} 
Download project settings from Polly

\item {} 
New Alignment algorithm: Obi-warp

\item {} 
Support for cdf files on Windows

\item {} 
Highlight detected compounds in the database

\end{itemize}

\item {} 
Enhancement
\begin{itemize}
\item {} 
Consistent sample ordering across widgets

\item {} 
Append Sample number to Sample Name for mzML files

\end{itemize}

\item {} 
Refactor
\begin{itemize}
\item {} 
Isotope Widget

\item {} 
Isotope detection

\end{itemize}

\item {} 
Bug fixes
\begin{itemize}
\item {} 
Fixed Isotopes expected m/z in mzroll

\item {} 
Blank samples will be visible in sample widget

\item {} 
Fix trailing delimiter in peak detailed format

\item {} 
Isotopic intensities are consistent between isotope widget and peaks table

\end{itemize}

\end{itemize}

\sphinxhref{https://zenodo.org/record/1168226}{Version 0.2.4}

Publication Date: February 7$^{\text{th}}$, 2018

\sphinxhref{https://zenodo.org/record/1165654}{Version 0.2.4}

Publication Date: February 5$^{\text{th}}$, 2018

\sphinxhref{https://zenodo.org/record/1158577}{Version 0.2.4 Beta}

Publication Date: January 24$^{\text{th}}$, 2018

\sphinxhref{https://zenodo.org/record/1157953}{Version 0.2.3}

Publication Date: January 23$^{\text{rd}}$, 2018

\sphinxhref{https://zenodo.org/record/1133506}{Version 0.2.2}

Publication Date: December 28$^{\text{th}}$, 2017


\section{Bugs and Feature Requests}
\label{\detokenize{BugsandFeatureRequests:bugs-and-feature-requests}}\label{\detokenize{BugsandFeatureRequests::doc}}
Existing bugs and feature requests can be found on El-MAVEN’s GitHub \sphinxhref{https://github.com/ElucidataInc/ElMaven/issues}{issue page}.
Please search the existing bugs and feature requests before you file one yourself.


\section{Copyright and License}
\label{\detokenize{CopyrightandLicense:copyright-and-license}}\label{\detokenize{CopyrightandLicense::doc}}
Code and documentation copyright 2017 \sphinxhref{http://www.elucidata.io/}{Elucidata Inc}. Code released under the \sphinxhref{https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html}{GPL v2.0}. Documentation is released under \sphinxhref{https://opensource.org/licenses/MIT}{MIT license}.



\renewcommand{\indexname}{Index}
\printindex
\end{document}